// @frontify/fondue/dist/components/Dialog/Dialog.es.js
import { jsx as D } from "react/jsx-runtime";
import { Modality as e } from "../../types/dialog.es.js";
import { Overlay as M } from "../../utilities/dialogs/Overlay.es.js";
import { Z_INDEX_MODAL as c } from "../../utilities/dialogs/constants.es.js";
import { Trigger as x } from "../../utilities/dialogs/Trigger.es.js";
import { Content as N } from "../../utilities/dialogs/Content.es.js";
const o = ({
  children: t,
  open: a,
  enablePortal: i = !0,
  maxHeight: d = "auto",
  maxWidth: m,
  minHeight: n = 200,
  minWidth: l = 600,
  modality: r = e.Modal,
  handleClose: f,
  "data-test-id": g = "fondue-dialog",
  darkUnderlay: p = !1,
  autoHeight: s = !1,
  verticalAlignment: u = "centered"
}) => /* @__PURE__ */ D(
  M,
  {
    "data-test-id": g,
    open: a,
    enablePortal: i,
    minWidth: l,
    minHeight: n,
    maxWidth: m,
    maxHeight: d,
    modality: r,
    handleClose: f,
    role: r === e.NonModal ? "region" : "dialog",
    darkUnderlay: p,
    autoHeight: s,
    zIndex: c,
    isDetached: !0,
    verticalAlignment: u,
    children: t
  }
);
o.displayName = "FondueDialog";
o.Trigger = x;
o.Content = N;
export {
  o as Dialog
};
