// @frontify/fondue/dist/components/RichTextEditor/Plugins/AutoformatPlugin/index.es.js
import { Plugin as i } from "../Plugin.es.js";
import { AUTOFORMAT_PLUGIN as p } from "./id.es.js";
import { createAutoformatPlugin as u, autoformatPunctuation as c, autoformatArrow as n, autoformatLegal as s, autoformatSubscriptNumbers as l, autoformatSubscriptSymbols as d, autoformatSuperscriptNumbers as f, autoformatSuperscriptSymbols as y, autoformatFraction as g, MARK_UNDERLINE as h, MARK_BOLD as E, MARK_ITALIC as T, MARK_STRIKETHROUGH as b, MARK_CODE as k, ELEMENT_UL as r, unwrapList as m, toggleList as e, ELEMENT_OL as a } from "@udecode/plate";
import { TextStyles as o } from "../TextStylePlugin/types.es.js";
import { ELEMENT_CHECK_ITEM as L } from "../CheckboxListPlugin/id.es.js";
class K extends i {
  constructor(t) {
    super(p, {
      ...t
    });
  }
  plugins() {
    return [
      u({
        options: {
          rules: [
            ...c,
            ...n,
            ...s,
            ...l,
            ...d,
            ...f,
            ...y,
            ...g,
            {
              mode: "mark",
              type: [h],
              match: ["_", "_"],
              ignoreTrim: !0
            },
            {
              mode: "mark",
              type: [E],
              match: ["**", "**"],
              ignoreTrim: !0
            },
            {
              mode: "mark",
              type: [T],
              match: ["*", "*"],
              ignoreTrim: !0
            },
            {
              mode: "mark",
              type: [b],
              match: ["~~", "~~"],
              ignoreTrim: !0
            },
            {
              mode: "mark",
              type: [k],
              match: ["`", "`"],
              ignoreTrim: !0
            },
            {
              mode: "block",
              match: ["# "],
              type: o.heading1
            },
            {
              mode: "block",
              match: ["## "],
              type: o.heading2
            },
            {
              mode: "block",
              match: ["### "],
              type: o.heading3
            },
            {
              mode: "block",
              match: ["#### "],
              type: o.heading4
            },
            {
              mode: "block",
              type: r,
              match: ["- "],
              preFormat: m,
              format: (t) => e(t, { type: r })
            },
            {
              mode: "block",
              type: a,
              match: ["1. ", "1) "],
              preFormat: m,
              format: (t) => e(t, { type: a })
            },
            {
              mode: "block",
              type: L,
              match: ["[] "]
            }
          ]
        }
      })
    ];
  }
}
export {
  K as AutoformatPlugin
};
