// @frontify/guideline-blocks-settings/dist/components/Link/LinkSelector/DocumentLink.es.js
import { jsxs as a, Fragment as l, jsx as e } from "react/jsx-runtime";
import { useState as b, useEffect as x } from "react";
import { PageLinks as m } from "./PageLinks.es.js";
import { merge as i, IconColorFan16 as p } from "@frontify/fondue";
const u = ({
  document: t,
  appBridge: c,
  selectedUrl: n,
  onSelectUrl: s,
  itemsToExpandInitially: r
}) => {
  const [o, w] = b(t.id === r.documentId), d = t.permanentLink === n;
  return x(() => {
    t.id === r.documentId && w(!0);
  }, [r, t.id]), /* @__PURE__ */ a(l, { children: [
    /* @__PURE__ */ a(
      "button",
      {
        "data-test-id": "internal-link-selector-document-link",
        className: i([
          "tw-flex tw-flex-1 tw-space-x-2 tw-items-center tw-py-2 tw-pr-2.5 tw-leading-5 tw-cursor-pointer tw-w-full",
          d ? "tw-bg-box-selected-strong tw-text-box-selected-strong-inverse hover:tw-bg-box-selected-strong-hover:hover hover:tw-text-box-selected-strong-inverse-hover:hover" : "hover:tw-bg-box-neutral-hover hover:tw-text-box-neutral-inverse-hover"
        ]),
        onClick: () => s(t.permanentLink),
        children: [
          /* @__PURE__ */ e(
            "button",
            {
              role: "button",
              tabIndex: 0,
              "data-test-id": "tree-item-toggle",
              className: "tw-flex tw-items-center tw-justify-center -tw-mr-2 tw-pr-3.5 tw-pt-1.5 tw-pb-1.5 tw-pl-3.5 tw-cursor-pointer",
              onClick: () => w(!o),
              children: /* @__PURE__ */ e(
                "div",
                {
                  className: i([
                    "tw-transition-transform tw-w-0 tw-h-0 tw-font-normal tw-border-t-4 tw-border-t-transparent tw-border-b-4 tw-border-b-transparent tw-border-l-4 tw-border-l-x-strong",
                    o ? "tw-rotate-90" : ""
                  ])
                }
              )
            }
          ),
          /* @__PURE__ */ e(p, {}),
          /* @__PURE__ */ e("span", { className: "tw-text-s", children: t.title }),
          /* @__PURE__ */ e("span", { className: "tw-flex-auto tw-font-sans tw-text-xs tw-text-right", children: "Document" })
        ]
      }
    ),
    o && /* @__PURE__ */ e(
      m,
      {
        appBridge: c,
        documentId: t.id,
        selectedUrl: n,
        onSelectUrl: s,
        itemsToExpandInitially: r
      }
    )
  ] });
};
export {
  u as DocumentLink
};
