// @frontify/guideline-blocks-settings/dist/components/Link/LinkSelector/PageLink.es.js
import { jsxs as l, Fragment as b, jsx as e } from "react/jsx-runtime";
import { useDocumentSection as p } from "@frontify/app-bridge";
import { merge as d } from "@frontify/fondue";
import { useState as u, useEffect as f } from "react";
import { SectionLink as v } from "./SectionLink.es.js";
const j = ({ page: t, selectedUrl: s, onSelectUrl: i, itemsToExpandInitially: r, appBridge: m }) => {
  const [n, w] = u(t.id === r.documentId), x = t.permanentLink === s, { documentSections: h } = p(m, t.id), o = [...h.values()], c = o.length > 0;
  return f(() => {
    t.id === r.pageId && w(!0);
  }, [r, t.id]), /* @__PURE__ */ l(b, { children: [
    /* @__PURE__ */ e(
      "button",
      {
        "data-test-id": "internal-link-selector-page-link",
        className: d([
          "tw-py-2 tw-pr-2.5 tw-leading-5 tw-cursor-pointer tw-flex tw-w-full",
          c ? "tw-pl-7" : "tw-pl-12",
          x ? "tw-bg-box-selected-strong tw-text-box-selected-strong-inverse hover:tw-bg-box-selected-strong-hover:hover hover:tw-text-box-selected-strong-inverse-hover:hover" : "hover:tw-bg-box-neutral-hover hover:tw-text-box-neutral-inverse-hover"
        ]),
        onClick: () => i(t.permanentLink),
        children: /* @__PURE__ */ l("div", { className: "tw-flex tw-flex-1 tw-space-x-1 tw-items-center tw-h-6", children: [
          c && /* @__PURE__ */ e(
            "button",
            {
              "data-test-id": "tree-item-toggle",
              className: "tw-flex tw-items-center tw-justify-center -tw-mr-2 tw-pr-3.5 tw-pt-1.5 tw-pb-1.5 tw-pl-3.5 tw-cursor-pointer",
              onClick: () => w(!n),
              children: /* @__PURE__ */ e(
                "div",
                {
                  className: d([
                    "tw-transition-transform tw-w-0 tw-h-0 tw-font-normal tw-border-t-4 tw-border-t-transparent tw-border-b-4 tw-border-b-transparent tw-border-l-4 tw-border-l-x-strong",
                    n ? "tw-rotate-90" : ""
                  ])
                }
              )
            }
          ),
          /* @__PURE__ */ e("span", { className: "tw-text-s", children: t.title }),
          /* @__PURE__ */ e("span", { className: "tw-flex-auto tw-font-sans tw-text-xs tw-text-right", children: "Page" })
        ] }, t.id)
      }
    ),
    n && o.length > 0 && o.map((a) => /* @__PURE__ */ e(
      v,
      {
        section: a,
        selectedUrl: s,
        onSelectUrl: i
      },
      a.id
    ))
  ] });
};
export {
  j as PageLink
};
