// @frontify/fondue/dist/components/LegacyTooltip/BrightHeader.es.js
import { jsx as r } from "react/jsx-runtime";
import { cloneElement as t } from "react";
import { merge as e } from "../../utilities/merge.es.js";
import { IconSize as n } from "../../foundation/Icon/IconSize.es.js";
import i from "../../foundation/Icon/Generated/IconExclamationMarkCircle.es.js";
import a from "../../foundation/Icon/Generated/IconExclamationMarkTriangle.es.js";
import s from "../../foundation/Icon/Generated/IconCheckMark.es.js";
import g from "../../foundation/Icon/Generated/IconDocument.es.js";
var m = /* @__PURE__ */ ((o) => (o.Information = "Information", o.Warning = "Warning", o.Tip = "Tip", o.Note = "Note", o))(m || {});
const b = {
  Information: "tw-bg-box-selected-strong",
  Warning: "tw-bg-box-negative-strong-pressed",
  Tip: "tw-bg-box-positive-strong-pressed",
  Note: "tw-bg-box-warning-strong"
}, v = {
  Information: "before:tw-bg-box-selected-strong",
  Warning: "before:tw-bg-box-negative-strong-pressed",
  Tip: "before:tw-bg-box-positive-strong-pressed",
  Note: "before:tw-bg-box-warning-strong"
}, c = {
  Information: /* @__PURE__ */ r(i, {}),
  Warning: /* @__PURE__ */ r(a, {}),
  Tip: /* @__PURE__ */ r(s, {}),
  Note: /* @__PURE__ */ r(g, {})
}, p = ({ headerStyle: o }) => /* @__PURE__ */ r(
  "div",
  {
    "data-test-id": "bright-header",
    className: e([
      "tw-h-9 tw-relative tw-z-[120000] tw-flex tw-justify-center tw-items-center tw-rounded-t-md tw-dark tw-text-text",
      b[o]
    ]),
    children: t(c[o], { size: n.Size20 })
  }
);
p.displayName = "FondueBrightHeader";
export {
  p as BrightHeader,
  m as BrightHeaderStyle,
  v as brightHeaderArrowBackgroundColors,
  b as brightHeaderBackgroundColors
};
