// @frontify/fondue/dist/components/LegacyTooltip/LegacyTooltip.es.js
import { jsxs as p, Fragment as ue, jsx as t } from "react/jsx-runtime";
import { IconSize as j } from "../../foundation/Icon/IconSize.es.js";
import { useLink as be } from "@react-aria/link";
import { FOCUS_VISIBLE_STYLE as ge } from "../../utilities/focusStyle.es.js";
import { merge as f } from "../../utilities/merge.es.js";
import { useState as g, useRef as D, useMemo as he, useCallback as h, useEffect as V, useLayoutEffect as Se, cloneElement as W } from "react";
import { BrightHeader as ve, brightHeaderArrowBackgroundColors as C } from "./BrightHeader.es.js";
import { usePopper as Ee } from "react-popper";
import { useMemoizedId as xe } from "../../hooks/useMemoizedId.es.js";
import { Button as Y } from "../Button/Button.es.js";
import { ButtonStyle as q, ButtonEmphasis as G, ButtonSize as J } from "../Button/ButtonTypes.es.js";
import { Portal as Le } from "../Portal/Portal.es.js";
const K = {
  small: "tw-pt-2",
  large: "tw-pt-3"
}, Q = {
  small: "tw-pb-2",
  large: "tw-pb-3"
};
var ke = /* @__PURE__ */ ((e) => (e.Top = "Top", e.Right = "Right", e.Bottom = "Bottom", e.Left = "Left", e))(ke || {}), ye = /* @__PURE__ */ ((e) => (e.Start = "Start", e.Middle = "Middle", e.End = "End", e))(ye || {});
const Ce = {
  "Top-Start": "top-start",
  "Top-End": "top-end",
  "Bottom-Start": "bottom-start",
  "Bottom-End": "bottom-end",
  "Left-Start": "left-start",
  "Left-End": "left-end",
  "Right-Start": "right-start",
  "Right-End": "right-end",
  "Top-Middle": "top",
  "Right-Middle": "right",
  "Bottom-Middle": "bottom",
  "Left-Middle": "left"
}, Te = (e, r, i) => {
  switch (!0) {
    case e.toString().includes("Top".toLowerCase()):
      return "before:tw-border-t-0 before:tw-border-l-0 tw-bottom-[-6px] before:tw-dark before:tw-bg-base";
    case e.toString().includes("Right".toLowerCase()):
      return f([
        "before:tw-border-t-0 before:tw-border-r-0 tw-left-[-5px]",
        r && i === "Start" && e.toString().includes("Start".toLowerCase()) ? C[r] : "before:tw-dark before:tw-bg-base"
      ]);
    case e.toString().includes("Bottom".toLowerCase()):
      return f([
        "before:tw-border-b-0 before:tw-border-r-0 tw-top-[-6px]",
        r ? C[r] : "before:tw-dark before:tw-bg-base"
      ]);
    case e.toString().includes("Left".toLowerCase()):
      return f([
        "before:tw-border-b-0 before:tw-border-l-0 tw-right-[-5px]",
        r && i === "Start" && e.toString().includes("Start".toLowerCase()) ? C[r] : "before:tw-dark before:tw-bg-base"
      ]);
    default:
      return "before:tw-border-b-0 before:tw-border-r-0 tw-top-[-6px] before:tw-dark before:tw-bg-base";
  }
}, Be = ({
  enablePortal: e = !1,
  children: r
}) => e ? /* @__PURE__ */ t(Le, { children: r }) : r, Re = ({
  content: e,
  tooltipIcon: r,
  heading: i,
  headingIcon: S,
  linkUrl: m,
  linkLabel: X,
  brightHeader: L,
  buttons: o,
  children: Z,
  position: T = "Bottom",
  alignment: B = "Middle",
  withArrow: v,
  flip: H = !0,
  triggerElement: P,
  hoverDelay: R = 200,
  enterDelay: k = 0,
  open: z = !1,
  disabled: A = !1,
  enablePortal: U = !1,
  hidden: ee = !1,
  "data-test-id": M = "tooltip"
}) => {
  var _;
  const [y, te] = g(
    null
  ), [s, d] = g(!1), N = D(null), n = ee || A, { linkProps: re } = be({ isDisabled: n }, N), oe = he(
    () => m || o || i || S,
    [m, o, i, S]
  ), se = Ce[`${T}-${B}`], [u, ae] = g(null), [E, ne] = g(null), [le, ie] = g(null), O = xe(), w = Ee(y, u, {
    placement: se,
    strategy: "fixed",
    modifiers: [
      {
        name: "arrow",
        options: {
          element: le
        }
      },
      {
        name: "eventListeners",
        options: { scroll: s, resize: s }
      },
      {
        name: "offset",
        options: {
          offset: [0, v ? 10 : 5]
        }
      },
      {
        name: "flip",
        enabled: H
      }
    ]
  }), de = ((_ = w.state) == null ? void 0 : _.placement) ?? T, fe = Te(de, L, B), l = D(null), we = h(() => {
    l.current && clearTimeout(l.current), l.current = setTimeout(() => d(!1), R);
  }, [R]), F = h(() => {
    if (l.current && clearTimeout(l.current), k) {
      l.current = setTimeout(() => d(!0), k);
      return;
    }
    d(!0);
  }, [k]), ce = h(
    (a) => {
      var c;
      const b = (c = a.composedPath) == null ? void 0 : c.call(a);
      b && [y, E, u].some(($) => $ && b.includes($)) && F();
    },
    [F, u, E, y]
  ), pe = h(
    (a) => {
      const { relatedTarget: b } = a;
      (!b || ![u, E].some((c) => c == null ? void 0 : c.contains(b))) && d(!1);
    },
    [u, E]
  ), I = n ? {} : {
    onMouseOver: (a) => ce(a.nativeEvent),
    onMouseLeave: we,
    onFocus: () => d(!0),
    onBlur: pe
  };
  V(() => {
    l.current && clearTimeout(l.current), d(n ? !1 : z);
  }, [z, n]);
  const x = h(
    (a) => {
      s && a.key === "Escape" && d(!1);
    },
    [s]
  );
  return Se(() => {
    typeof w.update == "function" && s && w.update();
  }, [s]), V(() => (s ? window.addEventListener("keydown", x) : window.removeEventListener("keydown", x), () => {
    window.removeEventListener("keydown", x);
  }), [x, s]), /* @__PURE__ */ p(ue, { children: [
    /* @__PURE__ */ t("div", { ...I, ref: ne, children: /* @__PURE__ */ t("div", { ref: te, "aria-labelledby": O, "aria-disabled": n, children: P }) }),
    /* @__PURE__ */ t(Be, { enablePortal: U, children: /* @__PURE__ */ p(
      "div",
      {
        ref: ae,
        "aria-hidden": n,
        className: f([
          "tw-popper-container tw-inline-block tw-max-w-[200px] tw-dark tw-bg-base tw-rounded-md tw-shadow-mid tw-text-text tw-z-[120000]",
          !s && "tw-opacity-0 tw-h-0 tw-w-0 tw-overflow-hidden"
        ]),
        "data-test-id": M,
        role: "tooltip",
        id: O,
        style: w.styles.popper,
        ...w.attributes.popper,
        ...I,
        children: [
          L && /* @__PURE__ */ t(ve, { headerStyle: L }),
          /* @__PURE__ */ p(
            "div",
            {
              className: f([
                "tw-px-3 tw-dark tw-bg-base tw-rounded-md tw-relative tw-z-[120000]",
                oe ? K.large : K.small,
                m ? Q.large : Q.small
              ]),
              children: [
                i && /* @__PURE__ */ p("h4", { className: "tw-flex tw-text-m tw-font-bold tw-mb-1", children: [
                  S && /* @__PURE__ */ t("span", { className: "tw-mr-1.5", children: W(S, { size: j.Size20 }) }),
                  i
                ] }),
                /* @__PURE__ */ p("div", { className: "tw-flex", children: [
                  r && /* @__PURE__ */ t("span", { className: "tw-shrink-0 tw-mr-1 tw-leading-4", children: W(r, { size: j.Size16 }) }),
                  /* @__PURE__ */ t("span", { className: "tw-text-s tw-min-w-0 tw-break-words", children: e })
                ] }),
                m && /* @__PURE__ */ t(
                  "a",
                  {
                    ...re,
                    "data-test-id": `${M}-link`,
                    ref: N,
                    href: m,
                    target: "_blank",
                    rel: "noopener noreferrer",
                    className: f([
                      "tw-text-xs tw-text-black-40 tw-underline tw-mt-1",
                      ge
                    ]),
                    children: X ?? "Click here to learn more."
                  }
                ),
                o && /* @__PURE__ */ p("div", { className: "tw-flex tw-flex-row-reverse tw-gap-x-1 tw-mt-4", children: [
                  o.length > 0 && /* @__PURE__ */ t(
                    Y,
                    {
                      style: q.Default,
                      emphasis: G.Strong,
                      size: J.Small,
                      onClick: o[0].action,
                      disabled: n,
                      children: o[0].label
                    }
                  ),
                  o.length === 2 && /* @__PURE__ */ t(
                    Y,
                    {
                      style: q.Default,
                      emphasis: G.Default,
                      size: J.Small,
                      onClick: o[1].action,
                      disabled: n,
                      children: o[1].label
                    }
                  )
                ] })
              ]
            }
          ),
          /* @__PURE__ */ t("div", { "aria-hidden": "true", children: /* @__PURE__ */ t(
            "div",
            {
              "data-test-id": "popover-arrow",
              "data-popper-arrow": v,
              ref: ie,
              style: w.styles.arrow,
              className: f([
                v && "tw-popper-arrow tw-z-[110000] tw-absolute tw-w-3 tw-h-3 tw-pointer-events-none before:tw-absolute before:tw-w-3 before:tw-h-3 before:tw-rotate-45 before:tw-border before:tw-border-line",
                v && fe
              ])
            }
          ) }),
          Z
        ]
      }
    ) })
  ] });
};
Re.displayName = "FondueLegacyTooltip";
export {
  Re as LegacyTooltip,
  ye as TooltipAlignment,
  ke as TooltipPosition
};
