// @frontify/fondue/dist/components/RichTextEditor/Plugins/ListPlugin/OrderedListPlugin/OrderedListMarkupElement.es.js
import { jsx as s } from "react/jsx-runtime";
import { usePlateEditorRef as c, ELEMENT_OL as l, findNodePath as p, getNodeAncestors as i, ELEMENT_LI as u } from "@udecode/plate";
import { MarkupElement as a } from "../../MarkupElement.es.js";
const m = [
  "[&>li>p]:before:tw-content-[counter(count,decimal)_'._']",
  "[&>li>p]:before:tw-content-[counter(count,_lower-alpha)_'._']",
  "[&>li>p]:before:tw-content-[counter(count,lower-roman)_'._']"
], d = (e, o) => {
  const t = p(e, o);
  if (!t)
    return 0;
  const r = i(e, t);
  return Array.from(r).filter((n) => n[0].type === u).length;
}, f = (e) => `tw-list-none tw-pl-[10px] tw-mb-[10px] tw-ml-[15px] [&>li>p]:before:tw-pr-1 [&>li>p]:before:tw-tabular-nums ${m[e % 3]}`, E = { counterReset: "count" }, L = ({
  attributes: e,
  children: o,
  element: t
}) => {
  const r = c(), n = d(r, t);
  return /* @__PURE__ */ s("ol", { className: f(n), ...e, style: E, children: o });
};
class N extends a {
  constructor(o = l, t = L) {
    super(o, t);
  }
}
export {
  E as OL_STYLES,
  N as OrderedListMarkupElement,
  L as OrderedListMarkupElementNode,
  f as getOrderedListClasses
};
