// @frontify/guideline-blocks-settings/dist/components/RichTextEditor/plugins/LinkPlugin/FloatingLink/EditLinkModal/useFloatingLinkEdit.es.js
import { useEditorRef as y, usePlateSelectors as F, useFloatingLinkSelectors as d, getPluginOptions as h, ELEMENT_LINK as n, getAboveNode as C, getPluginType as l, getRangeBoundingClientRect as R, getStartPoint as P, getEndPoint as b, getDefaultBoundingClientRect as N, useVirtualFloatingLink as O, floatingLinkActions as o, someNode as S, floatingLinkSelectors as c, useHotkeys as B, triggerFloatingLinkEdit as I, useFloatingLinkEnter as v, useFloatingLinkEscape as x, useComposedRef as A } from "@udecode/plate";
import { useCallback as D, useEffect as H } from "react";
import { getLinkFromEditor as T } from "../../../../../Link/utils/getUrl.es.js";
const M = ({ floatingOptions: g, ...i }) => {
  const e = y(), u = F().keyEditor(), a = d().mode(), f = d().isOpen(e.id), { triggerFloatingLinkHotkeys: k = "command+k, ctrl+k" } = h(e, n), m = D(() => {
    const t = C(e, {
      match: { type: l(e, n) }
    });
    if (t) {
      const [, s] = t;
      return R(e, {
        anchor: P(e, s),
        focus: b(e, s)
      });
    }
    return N();
  }, [e]), p = f && a === "edit", { update: r, style: E, floating: L } = O({
    editorId: e.id,
    open: p,
    getBoundingClientRect: m,
    ...g
  });
  return H(() => {
    const t = T(e);
    if (t && o.url(t), e.selection && S(e, {
      match: { type: l(e, n) }
    })) {
      o.show("edit", e.id), r();
      return;
    }
    c.mode() === "edit" && o.hide();
  }, [e, u, r]), B(
    k,
    (t) => {
      t.preventDefault(), c.mode() === "edit" && I(e);
    },
    {
      enableOnContentEditable: !0
    },
    []
  ), v(), x(), {
    style: {
      ...E,
      zIndex: 1e3
    },
    ...i,
    ref: A(i.ref, L)
  };
};
export {
  M as useFloatingLinkEdit
};
