// @frontify/fondue/dist/components/DialogFooter/DialogFooter.es.js
import { jsxs as l, jsx as o } from "react/jsx-runtime";
import { dialogPaddingMap as d } from "../../types/dialog.es.js";
import p from "../../foundation/Icon/Generated/IconArrowLeft.es.js";
import { merge as f } from "../../utilities/merge.es.js";
import { Box as e } from "../Box/Box.es.js";
import { Button as m } from "../Button/Button.es.js";
import { ButtonEmphasis as c, ButtonSize as w } from "../Button/ButtonTypes.es.js";
const u = ({
  actionButtons: a,
  children: s,
  backButton: r,
  padding: n = "small",
  "data-test-id": t = "fondue-dialog-footer"
}) => /* @__PURE__ */ l(
  e,
  {
    "data-test-id": t,
    className: f([
      "tw-border-t tw-border-t-line tw-flex tw-items-center tw-justify-between tw-gap-x-3",
      d[n]
    ]),
    children: [
      r && /* @__PURE__ */ o(
        m,
        {
          "data-test-id": `${t}-back-button`,
          ...r,
          emphasis: c.Default,
          icon: /* @__PURE__ */ o(p, {})
        }
      ),
      /* @__PURE__ */ o(e, { className: "tw-grow", "data-test-id": `${t}-content`, children: s }),
      /* @__PURE__ */ o(e, { className: "tw-flex tw-gap-x-3", "data-test-id": `${t}-action-buttons`, children: a.map((i) => /* @__PURE__ */ o(m, { ...i, size: w.Medium }, `${t}-button-${i.children}`)) })
    ]
  }
);
u.displayName = "FondueDialogFooter";
export {
  u as DialogFooter
};
