// @frontify/app-bridge/dist/tests/AppBridgeBlockStub.es.js
import F from "mitt";
import { spy as f, stub as e } from "sinon";
import { AssetDummy as v } from "./AssetDummy.es.js";
import { UserDummy as L } from "./UserDummy.es.js";
import { ColorPaletteDummy as o } from "./ColorPaletteDummy.es.js";
import { ColorDummy as l } from "./ColorDummy.es.js";
import { BulkDownloadDummy as w } from "./BulkDownloadDummy.es.js";
import { TemplateDummy as E } from "./TemplateDummy.es.js";
const u = 3452, K = 2341, O = 4561, B = 345214, J = ({
  blockSettings: D = {},
  blockAssets: I = {},
  editorState: k = !1,
  openAssetChooser: y = () => null,
  closeAssetChooser: C = () => null,
  blockId: p = u,
  sectionId: P = K,
  projectId: T = B,
  user: h = L.with(O),
  language: A = "en",
  privacySettings: S = {
    assetViewerEnabled: !1,
    assetDownloadEnabled: !1
  },
  blockTemplates: b = {}
} = {}) => {
  window.emitter = f(F()), window.blockSettings ?? (window.blockSettings = {}), window.blockSettings[p] = D;
  const a = {}, n = {}, d = {}, c = {};
  return {
    getBlockId: e().returns(p),
    getSectionId: e().returns(P),
    getProjectId: e().returns(T),
    getEditorState: e().returns(k),
    getBlockSettings: e().resolves(window.blockSettings),
    getAvailablePalettes: e().resolves([
      o.with(678, "Palette 1"),
      o.with(427, "Palette 2"),
      o.with(679, "Palette 3")
    ]),
    getColorPalettes: e().resolves([
      o.with(678, "Palette 1"),
      o.with(427, "Palette 2"),
      o.with(679, "Palette 3")
    ]),
    createColorPalette: e().resolves(
      o.with(678)
    ),
    updateColorPalette: e().resolves(
      o.with(678)
    ),
    getColorsByIds: e().resolves([
      l.red(9834),
      l.green(342),
      l.yellow(9314)
    ]),
    getColorsByColorPaletteId: e().resolves([
      l.red(9834),
      l.green(342),
      l.yellow(9314)
    ]),
    getColorPalettesWithColors: e().resolves([
      o.with(678, "Palette 1"),
      o.with(427, "Palette 2"),
      o.with(679, "Palette 3")
    ]),
    deleteColorPalette: e().resolves(),
    getAvailableColors: e().resolves([]),
    getCurrentLoggedUser: e().resolves(h),
    downloadColorKit: e().returns(
      `/api/color/export/${B}/zip/500`
    ),
    getAssetById: e().callsFake(
      (t) => Promise.resolve(v.with(t))
    ),
    closeAssetChooser: e().callsFake(() => {
      C();
    }),
    openAssetChooser: e().callsFake((t) => {
      y(t);
    }),
    getBlockAssets: e().callsFake(async () => Object.entries(I).reduce((t, [s, i]) => {
      const m = n[s] ?? [], g = a[s] ?? [];
      return t[s] = [
        ...i.filter((r) => !g.includes(r.id)),
        ...m.map((r) => v.with(r))
      ], t;
    }, {})),
    addAssetIdsToBlockAssetKey: e().callsFake(
      async (t, s) => {
        n[t] = [...n[t] ?? [], ...s];
      }
    ),
    deleteAssetIdsFromBlockAssetKey: e().callsFake(async (t, s) => {
      a[t] = [...a[t] ?? [], ...s];
    }),
    getBlockTemplates: e().callsFake(async () => Object.entries(b).reduce(
      (t, [s, i]) => {
        const m = c[s] ?? [], g = d[s] ?? [];
        return t[s] = [
          ...i.filter((r) => !g.includes(r.id)),
          ...m.map((r) => E.with(r))
        ], t;
      },
      {}
    )),
    addTemplateIdsToBlockTemplateKey: e().callsFake(async (t, s) => {
      c[t] = [...c[t] ?? [], ...s];
    }),
    deleteTemplateIdsFromBlockTemplateKey: e().callsFake(async (t, s) => {
      d[t] = [...d[t] ?? [], ...s];
    }),
    getTranslationLanguage: e().returns(A),
    getColors: e().resolves([
      l.red(9834),
      l.green(342),
      l.yellow(9314)
    ]),
    updateColor: e().callsFake(
      (t) => Promise.resolve(l.red(t))
    ),
    getBulkDownloadBySignature: e().resolves(
      w.default()
    ),
    getBulkDownloadByToken: e().resolves(
      w.default()
    ),
    getPrivacySettings: e().returns(S),
    api: e().withArgs().resolves().withArgs({
      name: "getAssetBulkDownloadToken",
      payload: { settingIds: ["settings1", "settings2"], documentBlockId: u }
    }).resolves({ assetBulkDownloadToken: "token" }).withArgs({
      name: "getAssetBulkDownloadToken",
      payload: { settingIds: void 0, documentBlockId: u }
    }).resolves({ assetBulkDownloadToken: "token" }),
    // TODO: Stub the following methods
    closeTemplateChooser: e(),
    openTemplateChooser: e(),
    createColor: e().resolves(l.red()),
    deleteColor: e().resolves(),
    getTemplateById: e().resolves({}),
    openAssetViewer: e(),
    updateBlockSettings: e().resolves(),
    getAllDocuments: e().resolves(),
    getUngroupedDocuments: e().resolves(),
    getDocumentsByDocumentGroupId: e().resolves(),
    getDocumentGroups: e().resolves(),
    getDocumentPagesByDocumentId: e().resolves(),
    getDocumentPagesByDocumentCategoryId: e().resolves(),
    getDocumentCategoriesByDocumentId: e().resolves(),
    getUncategorizedDocumentPagesByDocumentId: e().resolves(),
    getDocumentSectionsByDocumentPageId: e().resolves(),
    getDocumentTargets: e().resolves(),
    getDocumentPageTargets: e().resolves(),
    state: e().resolves(),
    context: e().resolves(),
    subscribe: e().resolves(),
    dispatch: e().resolves()
  };
};
export {
  J as getAppBridgeBlockStub
};
