// @frontify/guideline-blocks-settings/dist/components/Link/LinkInput.es.js
import { jsxs as b, jsx as t } from "react/jsx-runtime";
import { CheckboxState as l, FormControl as w, TooltipPosition as C, TextInput as S, ButtonSize as f, Checkbox as L } from "@frontify/fondue";
import { isValidUrlOrEmpty as N } from "./utils/url.es.js";
import { LinkSelector as O } from "./LinkSelector/LinkSelector.es.js";
const z = ({
  onUrlChange: i,
  onToggleTab: a,
  isValidUrlOrEmpty: o,
  appBridge: r,
  clearable: c,
  placeholder: d,
  newTab: s,
  openInNewTab: m,
  url: e = "",
  required: p,
  info: n,
  label: h,
  buttonSize: u,
  hideInternalLinkButton: x
}) => {
  const v = o ? o(e) : N(e), k = s ?? (m ? l.Checked : l.Unchecked);
  return /* @__PURE__ */ b("div", { "data-test-id": "link-input", children: [
    /* @__PURE__ */ t(
      w,
      {
        label: {
          children: h,
          htmlFor: "url",
          required: p,
          tooltip: n ? { content: n, position: C.Top } : void 0
        },
        children: /* @__PURE__ */ t(
          S,
          {
            id: "url",
            value: e,
            clearable: c,
            onChange: i,
            placeholder: d ?? "https://example.com",
            focusOnMount: !0
          }
        )
      }
    ),
    !v && /* @__PURE__ */ t("div", { className: "tw-text-text-negative tw-mt-1 tw-text-s", children: "Please enter a valid URL." }),
    !x && /* @__PURE__ */ t("div", { className: "tw-mt-3", children: /* @__PURE__ */ t(
      O,
      {
        url: e,
        appBridge: r,
        onUrlChange: i,
        buttonSize: u ?? f.Medium
      }
    ) }),
    /* @__PURE__ */ t("div", { className: "tw-mt-3", children: /* @__PURE__ */ t(L, { value: "new-tab", label: "Open in new tab", state: k, onChange: a }) })
  ] });
};
export {
  z as LinkInput
};
