// @frontify/fondue/dist/components/Badge/Badge.es.js
import { jsxs as u, jsx as a } from "react/jsx-runtime";
import { IconSize as c } from "../../foundation/Icon/IconSize.es.js";
import { FOCUS_VISIBLE_STYLE as g } from "../../utilities/focusStyle.es.js";
import { merge as m } from "../../utilities/merge.es.js";
import { cloneElement as B } from "react";
import { BadgeStatusIcon as C } from "./BadgeStatusIcon.es.js";
import { getStyleClasses as j, getCircularSizeClasses as v, getSizeClasses as z } from "./helpers.es.js";
import { BadgeEmphasis as d, BadgeStyle as E } from "./types.es.js";
import $ from "../../foundation/Icon/Generated/IconCross.es.js";
const h = ({
  children: e,
  status: r,
  icon: i,
  style: x = E.Primary,
  size: w = "medium",
  emphasis: b = d.None,
  disabled: s = !1,
  onClick: n,
  onDismiss: l,
  "data-test-id": o = "badge",
  withTitleAttribute: S = !0
}) => {
  if (!e && !i && !r)
    return null;
  const y = n ? "button" : "span", f = (t) => ["string", "number"].includes(typeof t) ? t : t instanceof Array ? t.map(f).join("") : t && typeof t == "object" && "props" in t ? f(t.props.children) : "", p = f(e), N = !e && !l && !!(r || i);
  return /* @__PURE__ */ u(
    "div",
    {
      className: m([
        "tw-relative tw-rounded-full tw-inline-flex tw-items-center tw-align-top tw-min-w-0 tw-flex-initial tw-transition-color",
        s ? "tw-bg-box-disabled tw-text-box-disabled-inverse" : j(x, !!n, b === d.Strong)
      ]),
      "data-test-id": o,
      children: [
        /* @__PURE__ */ u(
          y,
          {
            onClick: () => n && n(),
            className: m([
              "tw-inline-flex tw-items-center tw-justify-center tw-rounded-full tw-select-none tw-flex-initial tw-min-w-0",
              n && !s ? "hover:tw-cursor-pointer" : "tw-cursor-default",
              N ? v(w) : z(e, r, i, w, !!l),
              g
            ]),
            "data-test-id": `${o}-button`,
            title: S ? p : void 0,
            children: [
              r && /* @__PURE__ */ a(C, { status: r, disabled: s, "data-test-id": o }),
              i && /* @__PURE__ */ a(
                "span",
                {
                  "data-test-id": `${o}-icon`,
                  className: m([
                    "tw-flex-none tw-leading-none",
                    s && "tw-opacity-30",
                    !!e && "tw-pr-1"
                  ]),
                  children: B(i, { size: c.Size16 })
                }
              ),
              e && /* @__PURE__ */ a("span", { className: "tw-text-center tw-text-xxs tw-font-sans tw-font-normal tw-truncate tw-px-0.5", children: e })
            ]
          }
        ),
        l && /* @__PURE__ */ a(
          "button",
          {
            type: "button",
            "data-test-id": `${o}-dismiss`,
            className: m([
              "tw-absolute tw-rounded tw-leading-4",
              g,
              w === "medium" ? "tw-right-1.5" : "tw-right-1"
            ]),
            onClick: () => l(),
            "aria-label": `Dismiss ${p}`,
            children: /* @__PURE__ */ a($, { size: c.Size16 })
          }
        )
      ]
    }
  );
};
h.displayName = "FondueBadge";
export {
  h as Badge
};
