// @frontify/fondue/dist/components/RichTextEditor/Plugins/helper.es.js
import { jsx as e } from "react/jsx-runtime";
const r = ({ icon: t }) => /* @__PURE__ */ e("span", { className: "tw-p-2 tw-h-8 tw-justify-center tw-items-center tw-flex", children: t }), o = "tw-ml-0.5 !tw-text-text-weak hover:!tw-bg-box-selected hover:!tw-text-box-selected-inverse hover:tw-rounded", w = (t = !0) => ({
  root: t ? o : "!tw-text-text-weak !tw-cursor-not-allowed !tw-opacity-50",
  active: "!tw-bg-box-selected tw-rounded !tw-text-box-selected-inverse"
}), i = { root: { width: "24px", height: "24px" } }, c = ({ children: t, id: s }) => /* @__PURE__ */ e("div", { "data-plugin-id": s, children: t }), x = {
  left: "tw-text-left",
  right: "tw-text-right",
  center: "tw-text-center",
  justify: "tw-text-justify"
}, a = {
  left: "tw-justify-start tw-text-left",
  center: "tw-justify-center tw-text-center",
  justify: "tw-text-justify",
  right: "tw-justify-end tw-text-right"
}, l = "button";
export {
  c as ButtonWrapper,
  l as ELEMENT_BUTTON,
  r as IconStylingWrapper,
  x as alignmentClassnames,
  i as buttonStyles,
  w as getButtonClassNames,
  a as justifyClassNames
};
