// @frontify/fondue/dist/components/Grid/Grid.es.js
import { jsx as M } from "react/jsx-runtime";
import { Children as S } from "react";
import { GAP_Y_MAPPING as t, GAP_X_MAPPING as r, SPACING_VALUES as d, PADDING_VALUES_MAP as l, MARGIN_VALUES_MAP as n } from "../../utilities/dimensions.es.js";
import { merge as D } from "../../utilities/merge.es.js";
import { Box as E } from "../Box/Box.es.js";
const L = "fondue-grid", T = ({
  column: s = 2,
  spacingX: a = 4,
  spacingY: o = 4,
  children: m,
  "data-test-id": A = L,
  width: p = "100%",
  rowHeight: N = "auto",
  bg: _,
  color: G,
  margin: P = 0,
  padding: e = 0,
  as: c = "div"
}) => {
  const f = t[o] ? t[o] : t[4], u = r[a] ? r[a] : r[4], C = d.includes(e) ? l[e] : l[0], I = d.includes(e) ? n[P] : n[0];
  let i = 0;
  return S.map(m, (g) => (i++, g)), /* @__PURE__ */ M(
    E,
    {
      "data-test-id": A,
      className: D([
        "tw-grid",
        _,
        G,
        C,
        I,
        u,
        f
      ]),
      style: {
        width: p,
        gridTemplateColumns: `repeat(${s}, 1fr)`,
        gridTemplateRows: `repeat(${Math.ceil(i / s)}, ${N})`
      },
      as: c,
      children: m
    }
  );
};
T.displayName = "FondueGrid";
export {
  L as GRID_TEST_ID,
  T as Grid
};
