// @frontify/fondue/dist/components/DatePicker/DatePickerTrigger.es.js
import { jsxs as S, jsx as t } from "react/jsx-runtime";
import { merge as c } from "../../utilities/merge.es.js";
import { Validation as o } from "../../utilities/validation.es.js";
import { forwardRef as z } from "react";
import E from "../../foundation/Icon/Generated/IconCaretUp.es.js";
import F from "../../foundation/Icon/Generated/IconCaretDown.es.js";
import { TextInput as I } from "../TextInput/TextInput.es.js";
import O from "../../foundation/Icon/Generated/IconCalendar.es.js";
import { IconSize as l } from "../../foundation/Icon/IconSize.es.js";
const R = z(
  ({ value: e, onClick: i, isClearable: a, placeHolder: m, isCalendarOpen: f, validation: r = o.Default, onDateChanged: s }, d) => {
    const w = r === o.Error || r === o.Warning, g = "tw-right-14", p = "tw-right-8", h = "tw-right-4", n = ((u) => {
      switch (u) {
        case 0:
          return h;
        case 1:
          return p;
        case 2:
          return g;
      }
      return "";
    })(
      (e && a ? 1 : 0) + (w ? 1 : 0)
    );
    return /* @__PURE__ */ S("div", { ref: d, "aria-hidden": "true", children: [
      /* @__PURE__ */ t(
        "div",
        {
          className: c([
            "tw-absolute tw-top-2 tw-text-black-60 tw-z-[1]",
            n
          ]),
          children: f ? /* @__PURE__ */ t(E, { size: l.Size20 }) : /* @__PURE__ */ t(F, { size: l.Size20 })
        }
      ),
      /* @__PURE__ */ t(
        "div",
        {
          className: c([
            "tw-absolute tw-top-0 tw-left-0 tw-bottom-0 tw-z-10",
            n
          ]),
          onClick: i,
          "data-test-id": "open-close-click-zone"
        }
      ),
      /* @__PURE__ */ t(
        I,
        {
          decorator: /* @__PURE__ */ t(O, {}),
          placeholder: m,
          value: e,
          clearable: a,
          onEnterPressed: i,
          onClear: () => {
            s && s(null);
          },
          validation: r
        }
      )
    ] });
  }
);
R.displayName = "FondueDatePickerTrigger";
export {
  R as DatePickerTrigger
};
