// @frontify/fondue/dist/components/RichTextEditor/Plugins/CheckboxListPlugin/CheckboxListButton/CheckboxListToolbarButton.es.js
import { jsx as m } from "react/jsx-runtime";
import { usePlateEditorState as g, useEventPlateId as N, getNode as h, someNode as x, BlockToolbarButton as T, focusEditor as P, withoutNormalizing as y, unwrapList as C, toggleNodeType as L, getParentNode as b, setElements as E, ELEMENT_PARAGRAPH as i } from "@udecode/plate";
import { ELEMENT_CHECK_ITEM as k } from "../id.es.js";
const M = ({
  id: s,
  type: e = k,
  active: l,
  ...c
}) => {
  var n, p, u;
  const t = g(N(s)), a = ((n = t == null ? void 0 : t.selection) == null ? void 0 : n.focus.path) && h(t, (u = (p = t == null ? void 0 : t.selection) == null ? void 0 : p.focus) == null ? void 0 : u.path), o = l ?? (!!(t != null && t.selection) && x(t, { match: { type: e } }));
  return /* @__PURE__ */ m(
    T,
    {
      active: o,
      type: e,
      onClick: (f) => {
        f.preventDefault(), f.stopPropagation(), v(t, { type: e, isActive: o, node: a }), P(t);
      },
      ...c
    }
  );
}, v = (s, { type: e, isActive: l, node: c }) => y(s, () => {
  var t, a;
  if (s != null && s.selection) {
    if (l || C(s, {}), L(s, {
      activeType: e
    }), !(c != null && c.textStyle)) {
      const o = b(s, (a = (t = s == null ? void 0 : s.selection) == null ? void 0 : t.focus) == null ? void 0 : a.path), n = o && o[0].type;
      E(s, {
        textStyle: n
      });
    }
    l && E(s, {
      type: (c == null ? void 0 : c.textStyle) ?? i
    });
  }
});
export {
  M as CheckboxListToolbarButton,
  v as toggleCheckboxList
};
