// @frontify/fondue/dist/components/LinkChooser/LinkChooser.es.js
import { jsx as o, jsxs as h } from "react/jsx-runtime";
import { mapToAriaProps as pe } from "../ActionMenu/Aria/helper.es.js";
import { Checkbox as fe, CheckboxState as F } from "../Checkbox/Checkbox.es.js";
import { useDropdownAutoHeight as he } from "../../hooks/useDropdownAutoHeight.es.js";
import { useComboBox as we } from "@react-aria/combobox";
import { DismissButton as q } from "@react-aria/overlays";
import { scrollIntoView as Ce } from "@react-aria/utils";
import { useComboBoxState as Se } from "@react-stately/combobox";
import { Validation as ye } from "../../utilities/validation.es.js";
import { useMachine as Re } from "@xstate/react";
import { AnimatePresence as ge, motion as Ee } from "framer-motion";
import { useState as g, useMemo as Ie, useRef as w, useCallback as W, useEffect as E } from "react";
import { NavigationMenu as ke } from "./NavigationMenu.es.js";
import { Popover as De } from "./Popover.es.js";
import { SearchInput as be } from "./SearchInput.es.js";
import { SearchResultsList as xe } from "./SearchResultsList.es.js";
import { defaultSection as Be } from "./sections.es.js";
import { linkChooserMachine as Oe } from "./state/machine.es.js";
import { LinkChooserState as C } from "./state/types.es.js";
import { IconName as l } from "./types.es.js";
import { findSection as K, decoratedResults as ve, doesContainSubstring as Le, getDefaultData as Ne } from "./utils/helpers.es.js";
import { shouldGoBack as _e, isLoaded as Me, closeBoxState as U, queryMatchesSelection as Te, openBoxState as Pe } from "./utils/state.es.js";
import { createCustomLink as Ae } from "./utils/transformers.es.js";
import { useManualComboBoxEventHandlers as Fe } from "./utils/useManualComboBoxHandlers.es.js";
import qe from "../../foundation/Icon/Generated/IconDocument.es.js";
import We from "../../foundation/Icon/Generated/IconDocumentStack.es.js";
import Ke from "../../foundation/Icon/Generated/IconLink.es.js";
import Ue from "../../foundation/Icon/Generated/IconArrowOutExternal.es.js";
import $e from "../../foundation/Icon/Generated/IconLayersSingle.es.js";
import Ge from "../../foundation/Icon/Generated/IconBuildingBlock.es.js";
const Ve = {
  [l.Document]: /* @__PURE__ */ o(qe, {}),
  [l.Library]: /* @__PURE__ */ o(We, {}),
  [l.Link]: /* @__PURE__ */ o(Ke, {}),
  [l.External]: /* @__PURE__ */ o(Ue, {}),
  [l.Template]: /* @__PURE__ */ o($e, {}),
  [l.Block]: /* @__PURE__ */ o(Ge, {})
}, He = l.Link, It = "custom-link", kt = 5, Dt = "queries", bt = ({
  getGlobalByQuery: $ = Ne,
  openPreview: G = window.open,
  clipboardOptions: V = navigator.clipboard,
  selectedResult: d = null,
  openInNewTab: H = !1,
  ariaLabel: Q = "Menu",
  extraSections: c = [],
  label: I,
  placeholder: z,
  onOpenInNewTabChange: Y,
  onLinkChange: j,
  disabled: k,
  clearable: X,
  required: D,
  validation: J = ye.Default,
  "data-test-id": u = "link-chooser"
}) => {
  var P, A;
  const [{ context: t, matches: i, value: Z }, n, S] = Re(
    () => Oe.withContext({
      searchResults: [],
      selectedResult: d,
      query: "",
      interruptedFetch: !1,
      getExtraResultsByQuery: null,
      currentSectionId: Be.id,
      extraSections: c,
      clipboardOptions: V,
      openPreview: G,
      getGlobalByQuery: $,
      onLinkChange: j
    })
  ), [ee, b] = g(d == null ? void 0 : d.title), p = !_e(i), x = Ie(
    () => [
      !p && { id: "menu-top", menuItems: [K(c, t.currentSectionId)] },
      {
        id: "search",
        menuItems: ve(t.searchResults)
      },
      p && { id: "menu-bottom", menuItems: c.map(({ id: e, title: s }) => ({ id: e, title: s })) }
    ].filter(Boolean),
    [p, c, t.currentSectionId, t.searchResults]
  ), B = pe(Q, x), O = w(null), a = w(null), v = w(null), m = w(null), L = (e) => {
    const s = t.searchResults.find((de) => de.id === e);
    s && (b(s.title), n({ type: "SET_SELECTED_SEARCH_RESULT", data: { selectedResult: s } })), U(r), y(e);
  }, te = W(
    (e) => {
      b(e), n({ type: "TYPING", data: { query: e } });
    },
    [n]
  ), [N, y] = g((P = t.selectedResult) == null ? void 0 : P.id), r = Se({
    ...B,
    defaultFilter: (e, s) => Le(e, s, c),
    inputValue: ee,
    onInputChange: te,
    onSelectionChange: L,
    menuTrigger: "manual",
    shouldCloseOnBlur: !1,
    allowsEmptyCollection: !0,
    selectedKey: N
  }), { inputProps: oe, listBoxProps: ne, labelProps: re } = we(
    {
      ...B,
      inputRef: a,
      listBoxRef: v,
      popoverRef: m,
      isRequired: D,
      placeholder: z
    },
    r
  ), se = W(() => {
    r.setInputValue(""), y(""), n({ type: "CLEARING", data: { query: "" } });
  }, [n, r]), _ = () => {
    n("OPEN_DROPDOWN"), Pe(r);
  }, ie = () => {
    var e;
    document.activeElement !== a.current && ((e = a.current) == null || e.focus()), _();
  }, ae = (e) => {
    i(C.Focused) && e.target !== a.current && e.preventDefault();
  }, f = () => {
    let e = null;
    t.query && (Te(t.selectedResult, t.query) ? e = t.selectedResult : e = Ae(r.inputValue)), n({ type: "CLOSE_DROPDOWN", data: { selectedResult: e } }), e && N !== e.id && y(e.id), U(r);
  }, le = Fe(
    { inputProps: oe, inputRef: a, popoverRef: m, state: r },
    {
      onOpen: _,
      onClose: f,
      onNavigate: (e) => {
        var s;
        n(p ? {
          type: "SELECT_EXTRA_SECTION",
          data: {
            getExtraResultsByQuery: ((s = K(c, e)) == null ? void 0 : s.getResults) || null,
            currentSectionId: e.toString()
          }
        } : {
          type: "BACK_TO_DEFAULT",
          data: { getExtraResultsByQuery: null }
        });
      },
      onSelect: L
    }
  );
  E(() => {
    Me(i) && t.interruptedFetch && n({ type: "TYPING", data: { query: t.query } });
  }, [t.interruptedFetch, t.query, i, n, Z]);
  const { maxHeight: ce } = he(a, {
    isOpen: i(C.Focused),
    autoResize: !0
  }), {
    isOpen: M,
    selectionManager: { focusedKey: R }
  } = r;
  E(() => {
    if (R && m.current && M) {
      const e = m.current.querySelector(`[data-key="${R}"]`);
      e && Ce(m.current, e);
    }
  }, [R, M]);
  const [ue, me] = g(0), T = () => {
    var e;
    me(((e = O.current) == null ? void 0 : e.getBoundingClientRect().width) || 0);
  };
  return E(() => {
    T(), window.addEventListener("resize", T, !1);
  }, []), /* @__PURE__ */ h("div", { "data-test-id": u, ref: O, className: "tw-w-full tw-font-sans tw-text-s", children: [
    !!I && /* @__PURE__ */ h(
      "label",
      {
        ...re,
        "data-test-id": `${u}-label`,
        className: "tw-text-black-80 tw-mb-1 tw-flex tw-align-items-center",
        children: [
          I,
          D && /* @__PURE__ */ o(
            "span",
            {
              "data-test-id": `${u}-label-required`,
              className: "tw-h-4 tw-text-m tw-text-red-60 dark:tw-text-red-50 tw-ml-1",
              children: "*"
            }
          )
        ]
      }
    ),
    /* @__PURE__ */ o(
      be,
      {
        ariaProps: le,
        selectedResult: t.selectedResult,
        ref: a,
        disabled: k,
        decorator: Ve[((A = t.selectedResult) == null ? void 0 : A.icon) || He],
        clearable: X,
        onClear: se,
        machineService: S,
        validation: J,
        onClick: ie,
        onMouseDown: ae
      }
    ),
    /* @__PURE__ */ o(ge, { children: i(C.Focused) && /* @__PURE__ */ h(
      Ee.div,
      {
        style: {
          width: ue
        },
        className: "tw-absolute tw-left-auto tw-w-full tw-overflow-hidden tw-p-0 tw-shadow-mid tw-list-none tw-m-0 tw-mt-2 tw-z-20",
        initial: { height: 0 },
        animate: { height: "auto" },
        exit: { height: 0 },
        transition: { ease: [0.04, 0.62, 0.23, 0.98], duration: 0.5 },
        "data-test-id": `${u}-dropdown`,
        children: [
          /* @__PURE__ */ o(q, { onDismiss: f }),
          /* @__PURE__ */ h(
            De,
            {
              popoverRef: m,
              isOpen: i(C.Focused),
              onClose: f,
              maxHeight: ce,
              children: [
                /* @__PURE__ */ o(
                  xe,
                  {
                    ...ne,
                    listBoxRef: v,
                    state: r,
                    menuBlocks: x,
                    border: !1,
                    machineService: S
                  }
                ),
                /* @__PURE__ */ o("div", { "data-test-id": `${u}-action-menu`, className: "tw-border-t tw-border-black-10", children: /* @__PURE__ */ o(ke, { machineService: S, state: r }) })
              ]
            }
          ),
          /* @__PURE__ */ o(q, { onDismiss: f })
        ]
      },
      "content"
    ) }),
    /* @__PURE__ */ o("div", { className: "tw-my-2", "data-test-id": `${u}-new-tab`, children: /* @__PURE__ */ o(
      fe,
      {
        value: "new-tab",
        disabled: k,
        state: H ? F.Checked : F.Unchecked,
        onChange: Y,
        label: "Open in New Tab"
      }
    ) })
  ] });
};
export {
  It as CUSTOM_LINK_ID,
  He as DEFAULT_ICON,
  Ve as IconOptions,
  bt as LinkChooser,
  kt as MAX_STORED_ITEMS,
  Dt as QUERIES_STORAGE_KEY
};
