// @frontify/fondue/dist/components/LinkChooser/SearchResultOption.es.js
import { jsxs as c, jsx as t } from "react/jsx-runtime";
import { MenuItem as I } from "../MenuItem/MenuItem.es.js";
import { getInteractionModality as N } from "@react-aria/interactions";
import { useOption as y } from "@react-aria/listbox";
import { merge as j } from "../../utilities/merge.es.js";
import { useActor as F } from "@xstate/react";
import { useRef as L } from "react";
import { IconOptions as M } from "./LinkChooser.es.js";
import { DropdownState as w, LinkChooserState as O, SectionState as D } from "./state/types.es.js";
import { findSection as $ } from "./utils/helpers.es.js";
import { MenuItemContentSize as z } from "../MenuItem/types.es.js";
const Q = ({
  item: e,
  state: n,
  keyItemRecord: r,
  machineService: d
}) => {
  const a = L(null), { optionProps: p, isDisabled: f, isSelected: x, isFocused: u } = y(
    {
      key: e.key,
      shouldFocusOnHover: !1
    },
    n,
    a
  ), [
    {
      context: { extraSections: h, currentSectionId: b },
      matches: k
    }
  ] = F(d), s = r[e.key], g = s.icon ? M[s.icon] : void 0, o = $(h, b), l = (i) => k(`${O.Focused}.${i}.${D.Loaded}`), v = N() !== "pointer", S = (i) => {
    var m;
    return ((m = o == null ? void 0 : o.renderPreview) == null ? void 0 : m.call(o, i)) || /* @__PURE__ */ t(C, { ...i });
  };
  return /* @__PURE__ */ c(
    "li",
    {
      ...p,
      ref: a,
      className: j([
        "tw-relative hover:tw-bg-black-10 tw-list-none tw-outline-none",
        f && "tw-pointer-events-none tw-top-px",
        u && v && "tw-bg-black-10"
      ]),
      children: [
        l(w.Default) && /* @__PURE__ */ t(I, { ...s, active: x, decorator: g, size: z.Large }),
        l(w.ExtraSection) && S(s)
      ]
    }
  );
}, C = ({ title: e, subtitle: n, preview: r }) => /* @__PURE__ */ c("div", { className: "tw-flex tw-px-5 tw-py-1.5 tw-cursor-pointer", children: [
  /* @__PURE__ */ t("div", { className: "tw-flex tw-flex-shrink-0 tw-w-[75px] tw-h-[75px] tw-max-w-xs tw-bg-black-0 tw-rounded", children: r && /* @__PURE__ */ t("img", { className: "tw-w-full tw-object-contain", src: r, alt: e }) }),
  /* @__PURE__ */ c("div", { className: "tw-flex tw-flex-col tw-justify-center tw-ml-4", children: [
    /* @__PURE__ */ t("p", { className: "tw-block tw-text-md tw-leading-tight tw-hover:underline tw-text-black-80", children: e }),
    n && /* @__PURE__ */ t("p", { className: "tw-text-[12px] tw-text-black-80", children: n })
  ] })
] });
export {
  Q as SearchResultOption
};
