// @frontify/fondue/dist/components/LinkChooser/SearchResultsList.es.js
import { jsx as t, jsxs as n } from "react/jsx-runtime";
import { getMenuItems as k, getKeyItemRecord as N } from "../ActionMenu/Aria/helper.es.js";
import { useListBox as R } from "@react-aria/listbox";
import { useActor as v } from "@xstate/react";
import { useRef as I, useMemo as S } from "react";
import b from "./assets/background.svg.es.js";
import w from "./assets/no-results.svg.es.js";
import B from "./assets/nook-animated.png.es.js";
import { NavigationMenuItem as F } from "./NavigationMenu.es.js";
import { SearchResultSection as j } from "./SearchResultSection.es.js";
import { defaultSection as A } from "./sections.es.js";
import { findSection as E } from "./utils/helpers.es.js";
import { isFetching as L, isUnsuccessful as U, shouldGoBack as M } from "./utils/state.es.js";
const X = (e) => {
  const o = I(null), { listBoxRef: l = o, state: s, menuBlocks: d, machineService: a } = e, { listBoxProps: f } = R(e, s, l), h = k(d), p = N(h), [
    {
      context: { extraSections: m, currentSectionId: u, searchResults: x, query: c },
      matches: i
    },
    g
  ] = v(a), y = S(
    () => E(m, u) || A,
    [u, m]
  );
  return L(i) ? /* @__PURE__ */ t(T, {}) : U(i) ? /* @__PURE__ */ t(P, {}) : /* @__PURE__ */ n("div", { children: [
    M(i) && /* @__PURE__ */ t("div", { className: "tw-mt-2", children: /* @__PURE__ */ t(
      F,
      {
        state: s,
        section: y,
        onPress: () => {
          g({
            type: "BACK_TO_DEFAULT",
            data: { getExtraResultsByQuery: null }
          });
        },
        direction: "left"
      }
    ) }),
    /* @__PURE__ */ t(
      "ul",
      {
        ...f,
        "data-test-id": "link-chooser-results-list",
        ref: l,
        className: "tw-list-none tw-p-0 tw-m-0 tw-bg-white tw-z-20 focus-visible:tw-outline-none",
        children: x.length > 0 ? [...s.collection].filter((r) => r.key === "search").map((r) => /* @__PURE__ */ t(
          j,
          {
            heading: r,
            state: s,
            keyItemRecord: p,
            machineService: a
          },
          r.key
        )) : /* @__PURE__ */ t(
          K,
          {
            prompt: c ? `We could not find any results for "${c}".` : "Use the search above to discover your brand assets, or enter a URL",
            icon: c ? w : b
          }
        )
      }
    )
  ] });
}, K = ({ prompt: e, icon: o }) => /* @__PURE__ */ n(
  "div",
  {
    "data-test-id": "link-chooser-empty-results",
    className: "tw-flex tw-flex-col tw-justify-center tw-items-center tw-h-[278px] tw-p-3",
    children: [
      /* @__PURE__ */ t("img", { className: "tw-w-[150px] tw-mb-5", src: o, alt: "Icon" }),
      /* @__PURE__ */ t("p", { className: "tw-text-black-60 tw-text-center", children: e })
    ]
  }
), P = ({
  error: e = "An error occurred while fetching the results"
}) => /* @__PURE__ */ n(
  "div",
  {
    "data-test-id": "link-chooser-error",
    className: "tw-flex tw-flex-col tw-justify-center tw-items-center tw-h-[278px] tw-p-3",
    children: [
      /* @__PURE__ */ t("img", { className: "tw-w-[75px] tw-mb-5", src: w, alt: "Error" }),
      /* @__PURE__ */ t("p", { className: "tw-text-black-60 tw-text-center", children: e })
    ]
  }
), T = () => /* @__PURE__ */ t(
  "div",
  {
    "data-test-id": "link-chooser-loader",
    className: "tw-flex tw-flex-col tw-justify-center tw-items-center tw-h-[278px] tw-p-3",
    children: /* @__PURE__ */ t("img", { className: "tw-w-[50px]", src: B, alt: "Fetching" })
  }
);
export {
  X as SearchResultsList
};
