// @frontify/fondue/dist/components/Table/TableColumnHeader.es.js
import { jsx as e, jsxs as N } from "react/jsx-runtime";
import { CheckboxState as w, Checkbox as y } from "../Checkbox/Checkbox.es.js";
import { IconSize as D } from "../../foundation/Icon/IconSize.es.js";
import { FOCUS_VISIBLE_STYLE as E } from "../../utilities/focusStyle.es.js";
import { merge as m } from "../../utilities/merge.es.js";
import { useState as u, useRef as z, useEffect as B, cloneElement as U } from "react";
import d from "../../foundation/Icon/Generated/IconArrowBidirectional.es.js";
import { SortDirection as j, SelectionMode as F } from "./Table.es.js";
import L from "../../foundation/Icon/Generated/IconArrowDown.es.js";
import O from "../../foundation/Icon/Generated/IconArrowUp.es.js";
var _ = /* @__PURE__ */ ((t) => (t.Default = "Default", t.SelectAll = "SelectAll", t))(_ || {});
const H = ({
  column: t,
  type: f = "Default",
  rowIds: h,
  sortDirection: o,
  selectionMode: b,
  isColumnSorted: l = !1,
  handleSortChange: x,
  setSelectedRows: a
}) => {
  const {
    key: n,
    rendered: k,
    props: { allowsSorting: r }
  } = t, [S, c] = u(/* @__PURE__ */ e(d, {})), [i, g] = u(!1), s = z(null), C = r ? "button" : "span";
  if (B(() => {
    c(l ? o === j.Descending ? /* @__PURE__ */ e(L, {}) : /* @__PURE__ */ e(O, {}) : /* @__PURE__ */ e(d, {}));
  }, [l, o]), f === "SelectAll") {
    const p = {
      "aria-label": "Select all",
      scope: "col"
    }, I = i ? w.Checked : w.Unchecked, v = () => {
      a && (a(i ? [] : h), g((A) => !A));
    };
    return /* @__PURE__ */ e(
      "th",
      {
        ...p,
        ref: s,
        className: "tw-pl-8 tw-py-3 tw-pr-4 tw-w-16 tw-border-l-4 tw-border-transparent tw-group tw-outline-none",
        "data-test-id": "table-select-cell",
        children: b === F.SingleSelect ? /* @__PURE__ */ e("span", { className: "tw-sr-only", children: p["aria-label"] }) : /* @__PURE__ */ e(y, { value: n, "aria-label": n, state: I, onChange: v })
      }
    );
  }
  return /* @__PURE__ */ e(
    "th",
    {
      ref: s,
      className: m([
        "tw-text-xs tw-font-medium tw-text-black-100 dark:tw-text-white tw-px-4 tw-py-3 tw-outline-none tw-group focus-visible:bg-violet-90",
        r ? "tw-cursor-pointer" : ""
      ]),
      "data-test-id": "table-column",
      scope: "col",
      onClick: r ? () => x(t.key, o) : () => null,
      children: /* @__PURE__ */ N(
        C,
        {
          className: m([
            "tw-flex tw-gap-x-1 tw-items-center",
            E,
            r ? "tw-cursor-pointer" : "tw-cursor-default"
          ]),
          children: [
            k,
            r && /* @__PURE__ */ e(
              "span",
              {
                "aria-hidden": "true",
                className: l ? "tw-text-violet-50" : "tw-text-black-40 dark:tw-text-black-60 group-hover:tw-text-black-100 dark:group-hover:tw-text-white",
                children: U(S, { size: D.Size12 })
              }
            )
          ]
        }
      )
    }
  );
};
H.displayName = "FondueTableColumnHeader";
export {
  H as TableColumnHeader,
  _ as TableColumnHeaderType
};
