// @frontify/fondue/dist/components/Switch/Switch.es.js
import { jsxs as V, jsx as n } from "react/jsx-runtime";
import { useMemo as l } from "react";
import { merge as a } from "../../utilities/merge.es.js";
import { useFocusRing as j } from "@react-aria/focus";
import { FOCUS_STYLE as y } from "../../utilities/focusStyle.es.js";
import { useMemoizedId as W } from "../../hooks/useMemoizedId.es.js";
import { InputLabel as D } from "../InputLabel/InputLabel.es.js";
const U = "switch-container", L = {
  small: "tw-w-[22px] tw-h-3",
  medium: "tw-w-[30px] tw-h-4"
}, o = {
  small: {
    dimensions: "tw-w-3 tw-h-3",
    activeWidth: "group-active:tw-w-[14px]",
    activeTranslation: "tw-translate-x-[9px]"
  },
  medium: {
    dimensions: "tw-w-4 tw-h-4",
    activeWidth: "group-active:tw-w-[18px]",
    activeTranslation: "tw-translate-x-[13px]"
  }
}, p = "tw-w-2 tw-h-[1px]", T = ({
  id: C,
  name: f,
  label: d,
  disabled: e,
  onChange: h,
  size: s = "medium",
  mode: t = "off",
  labelStyle: v = "default",
  hug: b = !1,
  tooltip: g,
  "data-test-id": w = "switch"
}) => {
  const m = W(C), { isFocusVisible: x, focusProps: F } = j(), k = l(() => a(["tw-group tw-border tw-inline-flex tw-rounded-full tw-shrink-0 tw-p-0 tw-transition-colors", e && t === "indeterminate" ? "tw-flex tw-items-center tw-justify-center tw-bg-box-disabled tw-border-line tw-pointer-events-none" : e ? "tw-bg-box-disabled tw-border-line tw-pointer-events-none" : t === "indeterminate" ? "tw-bg-text-weak tw-flex tw-items-center tw-justify-center hover:tw-bg-text" : t === "on" ? "tw-bg-text-weak tw-border-line-xx-strong hover:tw-bg-text" : "tw-bg-box-neutral tw-border-line-x-strong hover:tw-bg-box-neutral-hover", L[s], x && y]), [t, e, s, x]), O = l(() => {
    const r = "tw-relative tw-self-center tw-transition-transform", c = t === "on" ? o[s].activeTranslation : "", i = e && t === "indeterminate" ? "tw-bg-text-disabled" : t === "indeterminate" ? "tw-bg-base" : c;
    return a([
      r,
      t === "indeterminate" ? p : o[s].dimensions,
      i
    ]);
  }, [e, t, s]), S = l(() => {
    const r = "tw-border tw-bg-base tw-rounded-full tw-absolute tw-block tw-self-center tw-transition-width tw-duration-75", u = t === "indeterminate" ? "tw-hidden" : t === "on" ? "tw-right-0  tw-translate-x-px" : "tw-left-0", i = e ? "tw-border-line-strong" : a(["tw-bg-base tw-border-line-xx-strong", o[s].activeWidth]);
    return a([
      r,
      t === "indeterminate" ? p : o[s].dimensions,
      u,
      i
    ]);
  }, [t, e, s]), N = l(() => a(["tw-gap-2 tw-items-center tw-justify-between", b ? "tw-inline-flex" : "tw-flex"]), [b]);
  return /* @__PURE__ */ V("div", { className: N, "data-test-id": `${w}-container`, children: [
    d && /* @__PURE__ */ n(D, { clickable: !0, htmlFor: m, disabled: e, tooltip: g, children: /* @__PURE__ */ n(
      "span",
      {
        "data-test-id": `${w}-label-wrapper`,
        className: v === "default" ? "tw-font-medium tw-text-text-weak" : "tw-font-bold tw-text-text",
        children: d
      }
    ) }),
    /* @__PURE__ */ n(
      "button",
      {
        ...F,
        id: m,
        disabled: e,
        name: f,
        "data-test-id": w,
        className: k,
        value: t,
        onClick: h,
        type: "button",
        children: /* @__PURE__ */ n("div", { className: O, children: /* @__PURE__ */ n("div", { className: S }) })
      }
    )
  ] });
};
T.displayName = "FondueSwitch";
export {
  U as SWITCH_ID,
  T as Switch
};
