// @frontify/fondue/dist/components/RichTextEditor/Plugins/LinkPlugin/FloatingLink/InsertLinkModal/InsertModal.es.js
import { jsxs as l, jsx as e } from "react/jsx-runtime";
import { Button as i } from "../../../../../Button/Button.es.js";
import { FormControl as o } from "../../../../../FormControl/FormControl.es.js";
import { Checkbox as x } from "../../../../../Checkbox/Checkbox.es.js";
import f from "../../../../../../foundation/Icon/Generated/IconCheckMark20.es.js";
import { TextInput as n } from "../../../../../TextInput/TextInput.es.js";
import { ButtonSize as a, ButtonStyle as b, ButtonEmphasis as v } from "../../../../../Button/ButtonTypes.es.js";
const j = ({
  state: t,
  onTextChange: d,
  onUrlChange: c,
  onToggleTab: m,
  onCancel: s,
  onSave: h,
  isValidUrlOrEmpty: r,
  hasValues: u,
  testId: w,
  children: p
}) => /* @__PURE__ */ l("div", { "data-test-id": w, className: "tw-bg-white tw-rounded tw-shadow tw-p-7 tw-min-w-[400px] tw-overflow-y-auto", children: [
  /* @__PURE__ */ e(
    o,
    {
      label: {
        children: "Text",
        htmlFor: "linkText",
        required: !0
      },
      children: /* @__PURE__ */ e(n, { id: "linkText", value: t.text, placeholder: "Link Text", onChange: d })
    }
  ),
  p,
  /* @__PURE__ */ l("div", { className: "tw-pt-5", children: [
    /* @__PURE__ */ e(
      o,
      {
        label: {
          children: "URL",
          htmlFor: "url",
          required: !0
        },
        children: /* @__PURE__ */ e(
          n,
          {
            id: "url",
            value: t.url,
            placeholder: "https://example.com",
            focusOnMount: !0,
            onChange: c
          }
        )
      }
    ),
    !r() && /* @__PURE__ */ e("div", { className: "tw-text-red-65 tw-mt-3", children: "Please enter a valid URL." })
  ] }),
  /* @__PURE__ */ e("div", { className: "tw-pt-5", children: /* @__PURE__ */ e(x, { value: "new-tab", label: "Open in new tab", state: t.newTab, onChange: m }) }),
  /* @__PURE__ */ e("div", { className: "tw-mt-3", children: /* @__PURE__ */ l("div", { className: "tw-pt-5 tw-flex tw-gap-x-3 tw-justify-end tw-border-t tw-border-t-black-10", children: [
    /* @__PURE__ */ e(
      i,
      {
        onClick: s,
        size: a.Medium,
        style: b.Default,
        emphasis: v.Default,
        children: "Cancel"
      }
    ),
    /* @__PURE__ */ e(
      i,
      {
        onClick: h,
        size: a.Medium,
        icon: /* @__PURE__ */ e(f, {}),
        disabled: !r() || !u,
        children: "Save"
      }
    )
  ] }) })
] });
export {
  j as InsertModal
};
