// @frontify/guideline-blocks-settings/dist/components/RichTextEditor/plugins/ButtonPlugin/components/FloatingButton/InsertButtonModal/InsertButtonModal.es.js
import { jsx as e, jsxs as d } from "react/jsx-runtime";
import { FormControl as m } from "@frontify/fondue";
import { useState as y } from "react";
import { InsertModal as f } from "../../../../LinkPlugin/FloatingLink/InsertLinkModal/InsertModal.es.js";
import { useInsertModal as p } from "./useInsertModal.es.js";
import { BlockStyles as n } from "../../../../styles.es.js";
const g = () => {
  const r = p(), { state: t, onButtonStyleChange: o } = r;
  return /* @__PURE__ */ e(f, { ...r, testId: "floating-button-insert", children: /* @__PURE__ */ e("div", { className: "tw-pt-5", children: /* @__PURE__ */ d(
    m,
    {
      label: {
        children: "Button Style",
        htmlFor: "buttonStyle",
        required: !0
      },
      children: [
        /* @__PURE__ */ e(
          i,
          {
            id: "primary",
            styles: n.buttonPrimary,
            isActive: t.buttonStyle === "primary",
            onClick: () => o("primary"),
            children: t.text || "Primary Button"
          }
        ),
        /* @__PURE__ */ e(
          i,
          {
            id: "secondary",
            styles: n.buttonSecondary,
            isActive: t.buttonStyle === "secondary",
            onClick: () => o("secondary"),
            children: t.text || "Secondary Button"
          }
        ),
        /* @__PURE__ */ e(
          i,
          {
            id: "tertiary",
            styles: n.buttonTertiary,
            isActive: t.buttonStyle === "tertiary",
            onClick: () => o("tertiary"),
            children: t.text || "Tertiary Button"
          }
        )
      ]
    }
  ) }) });
}, i = ({ id: r, styles: t, isActive: o, onClick: a, children: s }) => {
  const [u, l] = y(!1), c = () => t && t.hover && u ? { ...t, ...t.hover } : t;
  return /* @__PURE__ */ e(
    "button",
    {
      "data-test-id": `floating-button-insert-${r}`,
      onMouseEnter: () => l(!0),
      onMouseLeave: () => l(!1),
      onClick: a,
      style: { ...c(), marginTop: 0, marginBottom: 0 },
      className: o ? "tw-outline tw-outline-1 tw-outline-violet-60 tw-outline-offset-2 tw-w-fit" : "tw-w-fit",
      children: s
    }
  );
};
export {
  g as InsertButtonModal
};
