// @frontify/guideline-blocks-settings/dist/components/Attachments/Attachments.es.js
import { jsx as o, jsxs as p } from "react/jsx-runtime";
import { useState as r, useEffect as f } from "react";
import { useSensors as q, useSensor as U, PointerSensor as B, KeyboardSensor as G, DndContext as H, closestCenter as J, DragOverlay as Q } from "@dnd-kit/core";
import { SortableContext as X, rectSortingStrategy as Y, arrayMove as Z } from "@dnd-kit/sortable";
import { useEditorState as _, useAssetUpload as $ } from "@frontify/app-bridge";
import { LegacyTooltip as ee, TooltipPosition as te, Flyout as oe, FlyoutPlacement as se, IconPaperclip16 as ne, IconCaretDown12 as ie, AssetInput as re, AssetInputSize as le } from "@frontify/fondue";
import { SortableAttachmentItem as de, AttachmentItem as ae } from "./AttachmentItem.es.js";
import { restrictToWindowEdges as ce } from "@dnd-kit/modifiers";
const ye = ({
  items: l = [],
  onDelete: x,
  onReplaceWithBrowse: E,
  onReplaceWithUpload: O,
  onBrowse: W,
  onUpload: N,
  onSorted: P,
  appBridge: d
}) => {
  const [t, y] = r(l), [v, a] = r(!1), R = q(U(B), U(G)), [I, b] = r(void 0), [A, h] = r(!1), [c, u] = r([]), [m, T] = r(null), g = _(d), i = t == null ? void 0 : t.find((e) => e.id === I), [j, { results: S, doneAll: D }] = $({
    onUploadProgress: () => !A && h(!0)
  });
  f(() => {
    y(l);
  }, [l]), f(() => {
    m && (h(!0), j(m));
  }, [m]), f(() => {
    (async () => {
      D && (await N(S), h(!1));
    })();
  }, [D, S]);
  const k = () => {
    a(!1), d.openAssetChooser(
      (e) => {
        W(e), d.closeAssetChooser(), a(!0);
      },
      {
        multiSelection: !0,
        selectedValueIds: t.map((e) => e.id)
      }
    );
  }, C = (e) => {
    a(!1), d.openAssetChooser(
      async (s) => {
        a(!0), d.closeAssetChooser(), u([...c, e.id]), await E(e, s[0]), u(c.filter((n) => n !== e.id));
      },
      {
        multiSelection: !1,
        selectedValueIds: t.map((s) => s.id)
      }
    );
  }, F = async (e, s) => {
    u([...c, e.id]), await O(e, s), u(c.filter((n) => n !== e.id));
  }, z = (e) => {
    const { active: s } = e;
    b(s.id);
  }, V = (e) => {
    const { active: s, over: n } = e;
    if (n && s.id !== n.id && t) {
      const K = t.findIndex((w) => w.id === s.id), M = t.findIndex((w) => w.id === n.id), L = Z(t, K, M);
      y(L), P(L);
    }
    b(void 0);
  };
  return g || ((t == null ? void 0 : t.length) ?? 0) > 0 ? /* @__PURE__ */ o(
    ee,
    {
      withArrow: !0,
      position: te.Top,
      content: "Attachments",
      disabled: v,
      enterDelay: 500,
      triggerElement: /* @__PURE__ */ o("div", { "data-test-id": "attachments-flyout-button", children: /* @__PURE__ */ o(
        oe,
        {
          placement: se.BottomRight,
          onOpenChange: (e) => a(i ? !0 : e),
          isOpen: v,
          hug: !1,
          fitContent: !0,
          legacyFooter: !1,
          trigger: /* @__PURE__ */ p("div", { className: "tw-flex tw-text-[13px] tw-font-body tw-items-center tw-gap-1 tw-rounded-full tw-bg-box-neutral-strong-inverse hover:tw-bg-box-neutral-strong-inverse-hover active:tw-bg-box-neutral-strong-inverse-pressed tw-text-box-neutral-strong tw-outline tw-outline-1 tw-outline-offset-[1px] tw-p-[6px] tw-outline-line", children: [
            /* @__PURE__ */ o(ne, {}),
            /* @__PURE__ */ o("div", { children: l.length > 0 ? l.length : "Add" }),
            /* @__PURE__ */ o(ie, {})
          ] }),
          children: /* @__PURE__ */ p("div", { className: "tw-w-[300px]", children: [
            t.length > 0 && /* @__PURE__ */ p(
              H,
              {
                sensors: R,
                collisionDetection: J,
                onDragStart: z,
                onDragEnd: V,
                modifiers: [ce],
                children: [
                  /* @__PURE__ */ o(X, { items: t, strategy: Y, children: /* @__PURE__ */ o("div", { className: "tw-border-b tw-border-b-line", children: t.map((e) => /* @__PURE__ */ o(
                    de,
                    {
                      isEditing: g,
                      isLoading: c.includes(e.id),
                      item: e,
                      onDelete: () => x(e),
                      onReplaceWithBrowse: () => C(e),
                      onReplaceWithUpload: (s) => F(e, s)
                    },
                    e.id
                  )) }) }),
                  /* @__PURE__ */ o(Q, { children: i && /* @__PURE__ */ o(
                    ae,
                    {
                      isOverlay: !0,
                      isEditing: g,
                      item: i,
                      isDragging: !0,
                      onDelete: () => x(i),
                      onReplaceWithBrowse: () => C(i),
                      onReplaceWithUpload: (e) => F(i, e)
                    },
                    I
                  ) })
                ]
              }
            ),
            g && /* @__PURE__ */ p("div", { className: "tw-px-5 tw-py-3", children: [
              /* @__PURE__ */ o("div", { className: "tw-font-body tw-font-medium tw-text-text tw-text-s tw-my-4", children: "Add attachments" }),
              /* @__PURE__ */ o(
                re,
                {
                  isLoading: A,
                  size: le.Small,
                  onUploadClick: (e) => T(e),
                  onLibraryClick: k
                }
              )
            ] })
          ] })
        }
      ) })
    }
  ) : null;
};
export {
  ye as Attachments
};
