// @frontify/fondue/dist/components/LegacyOrderableList/LegacyOrderableList.es.js
import { jsx as s, jsxs as b } from "react/jsx-runtime";
import { useState as I, useEffect as g, Fragment as L } from "react";
import { LegacyCollectionItem as h } from "./LegacyCollectionItem.es.js";
import { HTML5Backend as y } from "react-dnd-html5-backend";
import { DndProvider as D } from "react-dnd";
import { useId as k } from "@react-aria/utils";
import { DropZone as d } from "../DropZone/DropZone.es.js";
import { moveItems as v } from "../../utilities/dnd/moveItems.es.js";
const C = (e, o) => e.sort === null && o.sort === null || e.sort === null ? 1 : o.sort === null ? -1 : e.sort - o.sort, F = ({
  onMove: e,
  dragDisabled: o,
  items: a,
  renderContent: m,
  "data-test-id": c = "orderable-list"
}) => {
  const [l, p] = I(a), n = k();
  g(() => {
    const t = [...a];
    t.sort(C), p(t);
  }, [a]);
  const i = (t, r, f) => {
    const u = v(t, r, f, l);
    e(u);
  };
  return /* @__PURE__ */ s(D, { backend: y, children: /* @__PURE__ */ s("div", { className: "tw-outline-none", "data-test-id": c, children: l.map((t, r) => /* @__PURE__ */ b(L, { children: [
    /* @__PURE__ */ s(
      d,
      {
        data: {
          targetItem: t,
          position: "before"
        },
        onDrop: i,
        accept: n
      },
      `orderable-list-item-${r}-before`
    ),
    /* @__PURE__ */ s(
      h,
      {
        dragDisabled: o,
        item: t,
        renderContent: m,
        listId: n
      },
      t.id
    ),
    r === a.length - 1 && /* @__PURE__ */ s(
      d,
      {
        data: {
          targetItem: t,
          position: "after"
        },
        onDrop: i,
        accept: n
      },
      `orderable-list-item-${r}-after`
    )
  ] }, `dropzone-orderable-list-key-${r}`)) }) });
};
F.displayName = "FondueLegacyOrderableList";
export {
  F as LegacyOrderableList
};
