// @frontify/fondue/dist/components/ColorInputFlyout/ColorPickerTrigger.es.js
import { jsx as o } from "react/jsx-runtime";
import { MenuItemContent as c } from "../MenuItem/MenuItemContent.es.js";
import { Trigger as g } from "../Trigger/Trigger.es.js";
import { IconSize as u } from "../../foundation/Icon/IconSize.es.js";
import { useMemoizedId as d } from "../../hooks/useMemoizedId.es.js";
import { useFocusRing as h } from "@react-aria/focus";
import { toShortRgb as x } from "../../utilities/colors.es.js";
import { merge as r } from "../../utilities/merge.es.js";
import { TinyColor as b } from "@ctrl/tinycolor";
import { ColorInputTitle as y } from "./ColorInputTitle.es.js";
import I from "../../foundation/Icon/Generated/IconDroplet.es.js";
const k = ({
  id: i,
  currentColor: t,
  format: m,
  isOpen: n = !1,
  disabled: e = !1,
  clearable: s = !1,
  onClear: w,
  onDelete: l
}) => {
  const { isFocusVisible: p, focusProps: f } = h(), a = t ? new b(x(t)).toRgbString() : "";
  return /* @__PURE__ */ o(
    g,
    {
      isOpen: n,
      disabled: e,
      isFocusVisible: p,
      showClear: !!t && s,
      onClear: w,
      onDelete: l,
      children: /* @__PURE__ */ o(
        "button",
        {
          ...f,
          type: "button",
          id: d(i),
          disabled: e,
          className: r([
            "tw-overflow-hidden tw-flex-auto tw-h-full tw-rounded tw-text-left tw-outline-none tw-py-2 tw-pl-3 tw-min-h-[34px] tw-pr-8",
            !t && "tw-text-black-60",
            e && "tw-text-black-40"
          ]),
          children: /* @__PURE__ */ o(
            c,
            {
              title: t ? /* @__PURE__ */ o(y, { format: m, currentColor: t }) : "Select color",
              decorator: t ? /* @__PURE__ */ o(
                "span",
                {
                  className: r([
                    "tw-h-4 tw-w-4 tw-rounded tw-flex tw-items-center tw-justify-center tw-ring-1 tw-ring-black-10 tw-ring-offset-1",
                    e && "tw-opacity-50"
                  ]),
                  style: { backgroundColor: a }
                }
              ) : /* @__PURE__ */ o("span", { className: "tw-text-black-70", children: /* @__PURE__ */ o(I, { size: u.Size12 }) })
            }
          )
        }
      )
    }
  );
};
k.displayName = "FondueColorInputTrigger";
export {
  k as ColorInputTrigger
};
