// @frontify/fondue/dist/components/Slider/Slider.es.js
import { jsxs as E, jsx as f } from "react/jsx-runtime";
import { useState as d, useCallback as X, useMemo as st, useEffect as A } from "react";
import { useMemoizedId as at } from "../../hooks/useMemoizedId.es.js";
import { merge as b } from "../../utilities/merge.es.js";
import { Validation as K } from "../../utilities/validation.es.js";
import { clamp as y } from "../../utilities/number.es.js";
import { debounce as lt } from "../../utilities/debounce.es.js";
import { TextInput as it, TextInputType as wt } from "../TextInput/TextInput.es.js";
var ct = /* @__PURE__ */ ((p) => (p.MinMax = "MinMax", p.ValueOutOfRange = "ValueOutOfRange", p.ValueNaN = "ValueNaN", p))(ct || {});
const ut = 0, ft = 100, dt = 1, pt = 5, ht = "Slider text input", bt = 3, B = ["ArrowUp", "ArrowRight"], S = ["ArrowDown", "ArrowLeft"], z = ["Home", "End"], mt = ({
  id: p,
  label: k,
  value: U,
  min: o = ut,
  max: a = ft,
  showMinMax: $ = !1,
  step: c = dt,
  stepMultiplier: Y = pt,
  valueSuffix: r = "",
  onError: D,
  onChange: I,
  "aria-label": P = ht,
  disabled: e = !1,
  "data-test-id": N = "fondue-slider"
}) => {
  const [l, m] = d(), [i, v] = d(""), [h, L] = d(), [T, R] = d(), [n, j] = d(null), [H, O] = d(!1), [F, x] = d(!1), _ = at(p), q = (t) => {
    v(`${t.replace(r, "")}${r}`);
  }, u = X(
    ({ clientX: t, rawValue: s }) => {
      if (!n)
        return;
      const w = a - o, et = n.getBoundingClientRect().x, ot = n.clientWidth;
      let V;
      if (t !== void 0)
        V = (t - et) / ot;
      else {
        V = ((s || o) - o) / w, R(y(V, 0, 1) * 100);
        return;
      }
      const rt = y(V, 0, 1), nt = w / c, W = y(Math.round(rt * nt) * c + o, o, a);
      R((W - o) / w * 100), v(`${W}${r}`);
    },
    [n, a, o, c, r]
  ), g = st(
    () => n ? lt((t) => {
      e || u({
        clientX: t.clientX ?? t.touches[0].clientX
      });
    }, bt) : () => {
    },
    [u, n, e]
  ), M = X(() => {
    n && (x(!1), n.removeEventListener("mousemove", g), n.addEventListener("touchmove", g), window.removeEventListener("mouseup", M));
  }, [n, g]), C = (t) => {
    !n || e || (x(!0), u({
      clientX: t.clientX ?? t.touches[0].clientX
    }), n.addEventListener("touchmove", g), n.addEventListener("mousemove", g), window.addEventListener("mouseup", M));
  }, G = (t) => {
    if (![...B, ...S, ...z].includes(t.key) || l === void 0 || e)
      return;
    t.preventDefault();
    let s = 0, w = 0;
    S.includes(t.key) ? s = -c : B.includes(t.key) && (s = c), z.includes(t.key) ? w = t.key === "Home" ? o : a : (s *= t.shiftKey ? Y : 1, w = y(Math.floor(l / c) * c + s, o, a)), m(w), v(`${w}${r}`);
  }, J = () => {
    e || O(!0);
  }, Q = () => {
    e || O(!1);
  }, Z = () => {
    e || x(!0);
  }, tt = () => {
    e || x(!1);
  };
  return A(() => {
    if (!n)
      return;
    if (o > a) {
      L(
        "MinMax"
        /* MinMax */
      );
      return;
    }
    const t = U || o;
    m(t), v(`${t}${r}`), u({ rawValue: t });
  }, [n, u, o, a, U, r]), A(() => {
    if (!i)
      return;
    const t = i.replace(r, ""), s = +t;
    if (Number.isNaN(s) || !t) {
      m(void 0), L(
        "ValueNaN"
        /* ValueNaN */
      );
      return;
    }
    if (s < o || s > a) {
      m(void 0), L(
        "ValueOutOfRange"
        /* ValueOutOfRange */
      );
      return;
    }
    L(void 0), m(s);
  }, [i, r, o, a]), A(() => {
    if (h || l === void 0 || !i)
      return;
    const s = +i.replace(r, "");
    Number.isNaN(s) || l !== s || (u({ rawValue: l }), e || I({
      raw: l,
      withSuffix: i.replace(r, "") + r
    }));
  }, [l, i, r, h, I, u, e]), A(() => {
    h && D && D(h);
  }, [h, D]), /* @__PURE__ */ E("div", { className: "tw-flex tw-flex-col", "data-test-id": N, id: _, children: [
    /* @__PURE__ */ f(
      "label",
      {
        htmlFor: _,
        className: b([!k && "tw-hidden", e && "tw-text-text-disabled"]),
        "data-test-id": `${N}-label`,
        children: k
      }
    ),
    /* @__PURE__ */ f("div", { className: "tw-flex", children: /* @__PURE__ */ E("div", { className: b(["tw-flex-1 tw-flex tw-items-center"]), children: [
      $ && /* @__PURE__ */ E("div", { className: b(["tw-mr-3", e && "tw-text-text-disabled"]), children: [
        o,
        r
      ] }),
      /* @__PURE__ */ E(
        "button",
        {
          ref: j,
          "data-test-id": `${N}-interactive`,
          className: "tw-flex-1 tw-relative tw-h-full tw-cursor-pointer disabled:tw-cursor-default tw-outline-none",
          onMouseOver: J,
          onMouseOut: Q,
          onFocus: Z,
          onBlur: tt,
          onMouseDown: C,
          onMouseUp: M,
          onTouchStart: C,
          onTouchEnd: M,
          onKeyDown: G,
          disabled: e,
          children: [
            /* @__PURE__ */ f("span", { className: "tw-absolute tw-block tw-top-1/2 tw--translate-y-1/2 tw-w-full tw-h-1 tw-rounded-sm tw-bg-box-neutral tw-flex-1" }),
            T !== void 0 && /* @__PURE__ */ f(
              "span",
              {
                role: "slider",
                "aria-valuenow": l,
                "aria-valuemin": o,
                "aria-valuemax": a,
                "aria-label": P,
                "data-test-id": `${N}-track`,
                className: b([
                  "tw-absolute tw-block tw-top-1/2 tw--translate-y-1/2 tw-origin-left tw-w-full tw-h-1  tw-rounded-sm tw-bg-box-neutral-strong tw-flex-1",
                  e && "tw-bg-box-neutral"
                ]),
                style: { width: `${T}%` }
              }
            ),
            /* @__PURE__ */ f(
              "span",
              {
                className: b([
                  "tw-absolute tw-block tw-top-1/2 tw--translate-y-1/2 tw--translate-x-1/2 tw-w-5 tw-h-5",
                  "after:tw-absolute after:tw-block after:tw-top-1/2 after:tw-left-1/2 after:tw--translate-y-1/2 after:tw--translate-x-1/2 after:tw-w-5 after:tw-h-5 after:tw-bg-base after:tw-rounded-full after:tw-border after:tw-border-line-strong",
                  H && !F && "after:tw-shadow",
                  F && "after:tw-border-line-xx-strong before:tw-content-[''] before:tw-opacity-25 before:tw-top-1/2 before:tw--translate-y-1/2 before:tw--translate-x-1/2 before:tw-left-1/2 before:tw-block before:tw-rounded-full before:tw-absolute before:tw-z-0 before:tw-w-9 before:tw-h-9 before:tw-bg-box-neutral-strong",
                  e && "after:tw-bg-box-neutral after:tw-border-line-weak"
                ]),
                style: { left: `${T}%` }
              }
            )
          ]
        }
      ),
      $ && /* @__PURE__ */ E("div", { className: b(["tw-ml-3", e && "tw-text-text-disabled"]), children: [
        a,
        r
      ] }),
      /* @__PURE__ */ f("div", { className: "tw-w-16 tw-ml-3", children: /* @__PURE__ */ f(
        it,
        {
          id: `${_}-text-input`,
          value: i,
          placeholder: P,
          type: wt.Text,
          validation: h ? K.Error : K.Default,
          onChange: q,
          disabled: e
        }
      ) })
    ] }) })
  ] });
};
mt.displayName = "FondueSlider";
export {
  ht as ARIA_LABEL_DEFAULT_VALUE,
  ft as MAX_DEFAULT_VALUE,
  ut as MIN_DEFAULT_VALUE,
  dt as STEP_DEFAULT_VALUE,
  pt as STEP_MULTIPLIER_DEFAULT_VALUE,
  mt as Slider,
  ct as SliderError
};
