// @frontify/fondue/dist/components/InlineDialog/InlineDialog.es.js
import { jsx as M } from "react/jsx-runtime";
import { Modality as r } from "../../types/dialog.es.js";
import { Trigger as x } from "../../utilities/dialogs/Trigger.es.js";
import { Content as N } from "../../utilities/dialogs/Content.es.js";
import { Overlay as y } from "../../utilities/dialogs/Overlay.es.js";
import { Z_INDEX_MODAL as C } from "../../utilities/dialogs/constants.es.js";
const o = ({
  children: e,
  open: i,
  placement: a = "bottom-start",
  offset: n = [0, 8],
  flip: l = !1,
  enablePortal: m = !0,
  maxHeight: d = "auto",
  maxWidth: f = 400,
  minHeight: g = 0,
  minWidth: s = 0,
  modality: t = r.Modal,
  handleClose: p,
  "data-test-id": u = "fondue-inlineDialog",
  darkUnderlay: D = !1,
  autoHeight: I = !1
}) => /* @__PURE__ */ M(
  y,
  {
    "data-test-id": u,
    open: i,
    placement: a,
    offset: n,
    flip: l,
    enablePortal: m,
    minWidth: s,
    minHeight: g,
    maxWidth: f,
    maxHeight: d,
    modality: t,
    handleClose: p,
    role: t === r.NonModal ? "region" : "dialog",
    darkUnderlay: D,
    autoHeight: I,
    zIndex: C,
    children: e
  }
);
o.displayName = "FondueInlineDialog";
o.Trigger = x;
o.Content = N;
export {
  o as InlineDialog
};
