// @frontify/fondue/dist/components/Toast/Toast.es.js
import { jsx as o, jsxs as l } from "react/jsx-runtime";
import { INSET_BORDER as c } from "../../utilities/borderStyle.es.js";
import { merge as a } from "../../utilities/merge.es.js";
import { AnimatePresence as w, motion as p } from "framer-motion";
import { toastStylesBackgroundColorsMap as T, ToastAnimationDirection as r } from "./types.es.js";
const S = "tw-min-h-[4.25rem] tw-bg-base tw-rounded-lg tw-shadow-mid tw-z-[120000]", y = a([S, c]), i = (t) => t === r.BottomToTop ? "150%" : "-150%", N = ({
  isOpen: t,
  style: s,
  icon: m,
  animationDirection: e = r.BottomToTop,
  children: n,
  "data-test-id": d = "toast"
}) => /* @__PURE__ */ o(w, { children: t && /* @__PURE__ */ o(
  p.div,
  {
    className: y,
    initial: {
      opacity: 0,
      y: i(e)
    },
    animate: { opacity: 1, y: 0 },
    exit: { opacity: 0, y: i(e) },
    "data-test-id": d,
    "aria-live": "polite",
    role: "alert",
    children: /* @__PURE__ */ l("div", { className: "tw-flex", children: [
      /* @__PURE__ */ o(
        "div",
        {
          className: a([
            "tw-flex tw-items-center tw-justify-center tw-min-h-[4.25rem] tw-min-w-[3rem] tw-rounded-l-lg tw-text-white",
            T[s]
          ]),
          children: m
        }
      ),
      n
    ] })
  }
) });
N.displayName = "FondueToast";
export {
  N as Toast
};
