// @frontify/fondue/dist/components/RichTextEditor/Plugins/CheckboxListPlugin/CheckboxListElement.es.js
import { jsxs as w, jsx as c } from "react/jsx-runtime";
import { ReactEditor as f } from "slate-react";
import { getRootProps as x, setNodes as E } from "@udecode/plate";
import { merge as i } from "../../../../utilities/merge.es.js";
import { MarkupElement as u } from "../MarkupElement.es.js";
import { ELEMENT_CHECK_ITEM as C } from "./id.es.js";
import { justifyClassNames as k } from "../helper.es.js";
import { useRichTextEditorContext as g } from "../../context/RichTextEditorContext.es.js";
const N = (t, e) => t[e.children[0].textStyle], S = "tw-flex tw-flex-row tw-pb-2 tw-gap-1.5 tw-items-center", b = "focus:tw-outline-none tw-w-auto tw-min-w-[10px]", y = (t) => {
  const { attributes: e, children: s, nodeProps: m, element: o, editor: n } = t, l = x(t), r = o.checked, a = o.align ?? "left", { styles: d } = g();
  return /* @__PURE__ */ w("div", { ...e, ...l, className: i([S, k[a]]), children: [
    /* @__PURE__ */ c("div", { contentEditable: !1, className: "tw-flex tw-items-center tw-justify-center tw-select-none", children: /* @__PURE__ */ c(
      "input",
      {
        "data-test-id": "checkbox-input",
        className: "tw-w-4 tw-h-4 tw-m-0",
        type: "checkbox",
        checked: !!r,
        onChange: (h) => {
          const p = f.findPath(n, o);
          E(n, { checked: h.target.checked }, { at: p });
        },
        ...m
      }
    ) }),
    /* @__PURE__ */ c(
      "span",
      {
        style: N(d, o),
        className: i([b, r && "!tw-line-through"]),
        children: s
      }
    )
  ] });
};
class R extends u {
  constructor(e = C, s = y) {
    super(e, s);
  }
}
export {
  S as CHECKBOX_DIV_CLASSES,
  b as CHECKBOX_SPAN_CLASSES,
  R as CheckboxListElement,
  y as CheckboxListElementNode
};
