// @frontify/fondue/dist/components/RichTextEditor/Plugins/CheckboxListPlugin/index.es.js
import { createPluginFactory as o, createIndentPlugin as r } from "@udecode/plate";
import { ELEMENT_CHECK_ITEM as e, CHECKBOX_LIST_PLUGIN as n } from "./id.es.js";
import { CheckboxListElement as i } from "./CheckboxListElement.es.js";
import { CHECKBOX_DIV_CLASSES as a, CHECKBOX_SPAN_CLASSES as k, CheckboxListElementNode as P } from "./CheckboxListElement.es.js";
import { CheckboxListButton as m } from "./CheckboxListButton/index.es.js";
import { Plugin as s } from "../Plugin.es.js";
const C = o({
  key: e,
  isElement: !0
});
class x extends s {
  constructor(t) {
    super(n, {
      button: m,
      markupElement: new i(),
      ...t
    });
  }
  plugins() {
    return [
      C(),
      r({
        inject: {
          props: {
            validTypes: [e]
          }
        }
      })
    ];
  }
}
export {
  a as CHECKBOX_DIV_CLASSES,
  n as CHECKBOX_LIST_PLUGIN,
  k as CHECKBOX_SPAN_CLASSES,
  i as CheckboxListElement,
  P as CheckboxListElementNode,
  x as CheckboxListPlugin,
  e as ELEMENT_CHECK_ITEM,
  C as createCheckboxListPlugin
};
