// @frontify/fondue/dist/components/RadioPill/RadioPill.es.js
import { jsxs as m, jsx as r } from "react/jsx-runtime";
import { IconSize as c } from "../../foundation/Icon/IconSize.es.js";
import { useFocusRing as n } from "@react-aria/focus";
import { FOCUS_STYLE as d } from "../../utilities/focusStyle.es.js";
import { merge as p } from "../../utilities/merge.es.js";
import { cloneElement as w } from "react";
const u = ({
  label: o,
  active: i,
  icon: t,
  onClick: e,
  "data-test-id": s = "radio-pill"
}) => {
  const { isFocusVisible: a, focusProps: l } = n();
  return /* @__PURE__ */ m(
    "button",
    {
      "data-test-id": s,
      type: "button",
      className: p([
        "tw-inline-flex tw-items-center tw-rounded-full tw-text-xs tw-px-2 tw-py-1",
        i ? "dark:tw-bg-black-5 dark:tw-text-black-100 tw-bg-black-100 tw-text-white" : "tw-text-black-80 dark:tw-text-black-40",
        a && d,
        e ? "tw-cursor-pointer" : "tw-cursor-auto"
      ]),
      onClick: e,
      ...l,
      children: [
        t && /* @__PURE__ */ r("span", { className: "tw-mr-1", children: w(t, { size: c.Size16 }) }),
        /* @__PURE__ */ r("span", { children: o })
      ]
    }
  );
};
u.displayName = "FondueRadioPill";
export {
  u as RadioPill
};
