// @frontify/guideline-blocks-settings/dist/components/BlockItemWrapper/Toolbar.es.js
import { jsx as e, jsxs as u } from "react/jsx-runtime";
import { LegacyTooltip as l, TooltipPosition as n, FOCUS_VISIBLE_STYLE as a, Flyout as v, IconDotsHorizontal16 as m, ActionMenu as f, MenuItemContentSize as x } from "@frontify/fondue";
import { DEFAULT_DRAGGING_TOOLTIP as g, DEFAULT_DRAG_TOOLTIP as y } from "./constants.es.js";
import { joinClassNames as c } from "../../utilities/react/joinClassNames.es.js";
const C = ({
  items: d,
  flyoutItems: s,
  isFlyoutOpen: b,
  setIsFlyoutOpen: w,
  isDragging: r,
  isFlyoutDisabled: p
}) => /* @__PURE__ */ e("div", { "data-test-id": "block-item-wrapper-toolbar", className: "tw-flex tw-justify-end", children: /* @__PURE__ */ u("div", { className: "tw-bg-white tw-isolate tw-text-box-selected-inverse tw-pointer-events-auto tw-flex tw-flex-shrink-0 tw-gap-[2px] tw-px-[1px] tw-spacing tw-items-center tw-h-7 tw-self-start tw-border tw-border-box-selected-inverse tw-rounded", children: [
  d.map(
    (t, o) => "draggableProps" in t ? /* @__PURE__ */ e(
      l,
      {
        withArrow: !0,
        hoverDelay: 0,
        enterDelay: 300,
        open: r,
        position: n.Top,
        content: /* @__PURE__ */ e("div", { children: r ? g : t.tooltip ?? y }),
        triggerElement: /* @__PURE__ */ e(
          "button",
          {
            ref: t.setActivatorNodeRef,
            "data-test-id": "block-item-wrapper-toolbar-btn",
            ...t.draggableProps,
            className: c([
              a,
              "tw-bg-base tw-relative tw-inline-flex tw-items-center tw-justify-center tw-w-6 tw-h-6 tw-rounded-sm focus-visible:tw-z-10",
              r ? "tw-cursor-grabbing tw-bg-box-selected-pressed" : "tw-cursor-grab hover:tw-bg-box-selected-hover"
            ]),
            children: t.icon
          }
        )
      },
      o
    ) : /* @__PURE__ */ e(
      l,
      {
        withArrow: !0,
        enterDelay: 300,
        hoverDelay: 0,
        disabled: r,
        position: n.Top,
        content: /* @__PURE__ */ e("div", { children: t.tooltip ?? "" }),
        triggerElement: /* @__PURE__ */ e(
          "button",
          {
            "data-test-id": "block-item-wrapper-toolbar-btn",
            onClick: t.onClick,
            className: c([
              "tw-bg-base tw-relative hover:tw-bg-box-selected-hover active:tw-bg-box-selected-pressed tw-cursor-pointer tw-inline-flex tw-items-center tw-justify-center tw-w-6 tw-h-6 tw-rounded-sm focus-visible:tw-z-10",
              a
            ]),
            children: t.icon
          }
        )
      },
      o
    )
  ),
  s.length > 0 && /* @__PURE__ */ e("div", { className: "tw-flex tw-flex-shrink-0 tw-flex-1 tw-h-6 tw-relative", children: /* @__PURE__ */ e(
    v,
    {
      isOpen: b && !r,
      isTriggerDisabled: p,
      legacyFooter: !1,
      fitContent: !0,
      hug: !1,
      onOpenChange: w,
      trigger: /* @__PURE__ */ e(
        l,
        {
          withArrow: !0,
          hoverDelay: 0,
          enterDelay: 300,
          disabled: r,
          position: n.Top,
          content: /* @__PURE__ */ e("div", { children: "Options" }),
          triggerElement: /* @__PURE__ */ e(
            "div",
            {
              "data-test-id": "block-item-wrapper-toolbar-flyout",
              className: "tw-bg-base hover:tw-bg-box-selected-hover active:tw-bg-box-selected-pressed tw-cursor-pointer tw-inline-flex tw-items-center tw-justify-center tw-w-6 tw-h-6 tw-rounded-sm tw-relative",
              children: /* @__PURE__ */ e(m, {})
            }
          )
        }
      ),
      children: /* @__PURE__ */ e(
        f,
        {
          menuBlocks: s.map((t, o) => ({
            id: o.toString(),
            menuItems: t.map((i, h) => ({
              id: o.toString() + h.toString(),
              size: x.XSmall,
              title: i.title,
              style: i.style,
              onClick: () => {
                w(!1), i.onClick();
              },
              initialValue: !0,
              decorator: /* @__PURE__ */ e("div", { className: "tw-mr-2", children: i.icon })
            }))
          }))
        }
      )
    }
  ) })
] }) });
export {
  C as Toolbar
};
