// @frontify/fondue/dist/components/DialogHeader/DialogHeader.es.js
import { jsxs as m, jsx as t } from "react/jsx-runtime";
import { merge as l } from "../../utilities/merge.es.js";
import { dialogPaddingMap as s } from "../../types/dialog.es.js";
import { FOCUS_VISIBLE_STYLE as c } from "../../utilities/focusStyle.es.js";
import { Flex as p } from "../Flex/Flex.es.js";
import { Text as w } from "../../typography/Text/Text.es.js";
import f from "../../foundation/Icon/Generated/IconCross.es.js";
import { Box as g } from "../Box/Box.es.js";
const b = ({
  title: d,
  collapseBottom: n = !1,
  size: r = "medium",
  onClose: o,
  icon: i,
  badge: a,
  "data-test-id": e = "fondue-dialog-header"
}) => /* @__PURE__ */ m(
  g,
  {
    "data-test-id": e,
    className: l([
      "tw-px-6 tw-flex tw-items-center tw-border-b tw-border-b-line",
      s[r],
      n && "tw-pb-0 tw-border-none"
    ]),
    children: [
      /* @__PURE__ */ m(p, { alignItems: "center", "data-test-id": `${e}-title`, children: [
        i && /* @__PURE__ */ t("span", { "data-test-id": `${e}-icon`, className: "tw-mr-1.5", children: i }),
        /* @__PURE__ */ t(w, { weight: "x-strong", size: r, children: d }),
        a && /* @__PURE__ */ t("div", { "data-test-id": `${e}-badge`, className: "tw-ml-1.5", children: a })
      ] }),
      o && /* @__PURE__ */ t(
        "button",
        {
          "data-test-id": `${e}-close`,
          "aria-label": `${e}-close`,
          onClick: o,
          className: l([c, "tw-rounded tw-absolute tw-right-5"]),
          children: /* @__PURE__ */ t(f, {})
        }
      )
    ]
  }
);
b.displayName = "FondueDialogHeader";
export {
  b as DialogHeader
};
