// @frontify/fondue/dist/components/ActionMenu/ActionMenu/ActionMenu.es.js
import { jsx as t } from "react/jsx-runtime";
import { AriaList as I } from "../Aria/AriaList.es.js";
import { AriaMenuItem as M } from "../Aria/AriaMenuItem.es.js";
import { AriaSection as b } from "../Aria/AriaSection.es.js";
import { getMenuItems as R, getKeyItemRecord as g, mapToAriaProps as h, getDisabledItemIds as k } from "../Aria/helper.es.js";
import { useMenu as K } from "@react-aria/menu";
import { useTreeState as L } from "@react-stately/tree";
import { useRef as P } from "react";
const x = ({
  menuBlocks: r,
  ariaLabel: a = "Action Menu",
  focus: c,
  border: p = !0,
  scrollable: u = !1,
  onClick: l
}) => {
  const n = R(r), d = g(n), i = h(a, r), e = L({
    ...i,
    selectionMode: "none",
    disabledKeys: k(n)
  }), s = P(null), { menuProps: f } = K({ ...i, autoFocus: c }, e, s);
  return /* @__PURE__ */ t(I, { ariaProps: { ...f }, ref: s, border: p, scrollable: u, children: [...e.collection].map((m) => {
    const { key: y, "aria-label": A } = m;
    return /* @__PURE__ */ t(b, { ariaLabel: A, children: [...m.childNodes].map((o) => /* @__PURE__ */ t(
      M,
      {
        menuItem: d[o.key],
        state: e,
        node: o,
        onClick: l
      },
      o.key
    )) }, y);
  }) });
};
x.displayName = "FondueActionMenu";
export {
  x as ActionMenu
};
