// @frontify/fondue/dist/components/Modal/ModalVisual.es.js
import { jsx as a } from "react/jsx-runtime";
import { merge as i } from "../../utilities/merge.es.js";
import { FrontifyPattern as m } from "../FrontifyPattern/FrontifyPattern.es.js";
import { patternThemes as s } from "../../foundation/Pattern/PatternTheme.es.js";
import { PatternScale as o } from "../../foundation/Pattern/PatternScale.es.js";
import { PatternDesign as l } from "../../foundation/Pattern/PatternDesigns.es.js";
const r = ({
  pattern: e = l.DigitalAssets,
  foregroundColor: t
}) => /* @__PURE__ */ a(
  "div",
  {
    "data-test-id": "modal-visual",
    className: i([
      "tw-h-[1802px] tw-absolute tw-bottom-0",
      t && s[t]
    ]),
    style: {
      clipPath: "path('M0 0L252 145.492C256.95 148.35 260 153.632 260 159.349V1641.98C260 1647.7 256.95 1652.98 252 1655.84L0 1801.33V1501.11V0Z')"
    },
    children: /* @__PURE__ */ a("div", { className: "tw-absolute tw-bottom-0", children: /* @__PURE__ */ a(
      m,
      {
        pattern: e,
        scaleOrigin: ["left", "bottom"],
        scale: o.XXL,
        foregroundColor: t
      }
    ) })
  }
);
r.displayName = "FondueModalVisual";
export {
  r as ModalVisual
};
