// @frontify/fondue/dist/components/Modal/ModalHeader.es.js
import { jsxs as r, jsx as t } from "react/jsx-runtime";
import { useContext as i, cloneElement as c } from "react";
import { merge as m } from "../../utilities/merge.es.js";
import { modalHeaderVariants as p, ModalHeaderVariant as f } from "./types.es.js";
import { IconSize as w } from "../../foundation/Icon/IconSize.es.js";
import { ModalTitle as x } from "./context/ModalTitle.es.js";
import { ModalLayout as h } from "./context/ModalLayout.es.js";
const u = ({
  title: d,
  leadText: e,
  decorator: a,
  variant: l = f.Default
}) => {
  const n = i(x), { padding: o, compact: s } = i(h);
  return /* @__PURE__ */ r("div", { "data-test-id": "modal-header", className: `${o.top} ${o.horizontal}`, children: [
    /* @__PURE__ */ r("div", { className: "tw-flex tw-items-center", children: [
      a && /* @__PURE__ */ t(
        "span",
        {
          "data-test-id": "modal-header-decorator-container",
          className: m(["tw-mr-2", p[l]]),
          children: c(a, { size: w.Size24 })
        }
      ),
      /* @__PURE__ */ t(
        "h3",
        {
          ...n,
          className: m([
            "tw-font-heading tw-font-medium tw-text-text",
            s ? "tw-text-lg" : "tw-text-xl"
          ]),
          children: d
        }
      )
    ] }),
    e && /* @__PURE__ */ t("p", { className: "tw-text-text-weak tw-mt-4", children: e })
  ] });
};
u.displayName = "FondueModalHeader";
export {
  u as ModalHeader
};
