// @frontify/fondue/dist/components/ActionMenu/Aria/AriaMenuItem.es.js
import { jsx as r } from "react/jsx-runtime";
import { MenuItem as A } from "../../MenuItem/MenuItem.es.js";
import { useFocusRing as k } from "@react-aria/focus";
import { useMenuItem as x } from "@react-aria/menu";
import { mergeProps as F } from "@react-aria/utils";
import { FOCUS_STYLE_INSET as P } from "../../../utilities/focusStyle.es.js";
import { merge as j } from "../../../utilities/merge.es.js";
import { useRef as E, useState as d, useEffect as N } from "react";
import { Switch as O } from "../../Switch/Switch.es.js";
const z = (o) => typeof o.onClick < "u" && typeof o.type > "u", c = (o) => typeof o.onClick < "u" && o.type === "switch", R = (o, e) => {
  const [t, l] = d(e ?? !1), [i, s] = d({
    switchComponent: void 0,
    switchValue: null,
    toggleSwitch: null
  });
  return N(() => {
    o && s({
      switchComponent: /* @__PURE__ */ r(O, { size: "small", mode: t ? "on" : "off" }),
      switchValue: t,
      toggleSwitch: () => l(!t)
    });
  }, [o, t]), i;
}, T = ({ menuItem: o, node: e, state: t, isSelected: l, onClick: i }) => {
  const s = E(null), a = c(o) ? o.initialValue : !1, {
    switchComponent: f = void 0,
    switchValue: w = null,
    toggleSwitch: n = null
  } = R(c(o), a), { menuItemProps: h } = x(
    {
      ...e,
      onAction: () => {
        z(o) && o.onClick(), c(o) && (n == null || n(), o.onClick(!w));
      },
      isDisabled: (o == null ? void 0 : o.disabled) || !1,
      "aria-label": typeof (o == null ? void 0 : o.title) == "string" ? o == null ? void 0 : o.title : "Menu item"
    },
    t,
    s
  ), { title: u, decorator: b, subtitle: g, size: S, style: C, disabled: p, selectionIndicator: y } = o, { isFocusVisible: v, focusProps: M } = k();
  return (
    // eslint-disable-next-line jsx-a11y/click-events-have-key-events,jsx-a11y/no-noninteractive-element-interactions
    /* @__PURE__ */ r(
      "li",
      {
        ...F(h, M),
        "data-test-id": "menu-item",
        className: j([
          "tw-relative hover:tw-bg-box-neutral-hover tw-list-none tw-outline-none",
          p && "tw-pointer-events-none tw-top-px",
          v && P
        ]),
        ref: s,
        onClick: (V) => {
          V.stopPropagation(), i && i();
        },
        children: /* @__PURE__ */ r(
          A,
          {
            title: u,
            decorator: b,
            subtitle: g,
            size: S,
            style: C,
            disabled: p,
            active: l,
            selectionIndicator: y,
            switchComponent: f
          }
        )
      }
    )
  );
};
T.displayName = "FondueAriaMenuItem";
export {
  T as AriaMenuItem
};
