// @frontify/app-bridge/dist/index.es.js
import { enableMapSet as e } from "immer";
import { AppBridgeCreateAsset as m } from "./AppBridgeCreateAsset.es.js";
import { AppBridgePlatformApp as s } from "./AppBridgePlatformApp.es.js";
import { HttpClientError as i } from "./errors/HttpClientError.es.js";
import { InitializationError as f } from "./errors/InitializationError.es.js";
import { FetchError as n } from "./errors/NotifyError.es.js";
import { TimeoutReachedError as l } from "./errors/TimeoutReachedError.es.js";
import { ValidationError as g } from "./errors/ValidationError.es.js";
import { createAsset as A } from "./registries/api/CreateAsset.es.js";
import { getAssetResourceInformation as T } from "./registries/api/GetAssetResourceInformation.es.js";
import { getCurrentUser as P } from "./registries/api/GetCurrentUser.es.js";
import { closeAssetChooser as h, openAssetChooser as B } from "./registries/commands/AssetChooser.es.js";
import { openAssetViewer as k } from "./registries/commands/AssetViewer.es.js";
import { downloadAsset as R } from "./registries/commands/DownloadAsset.es.js";
import { openNavigationManager as w } from "./registries/commands/NavigationManager.es.js";
import { closeTemplateChooser as G, openTemplateChooser as L } from "./registries/commands/TemplateChooser.es.js";
import { openNewPublication as j } from "./registries/commands/NewPublication.es.js";
import { openConnection as I } from "./registries/commands/Connection.es.js";
import { AssetBulkDownloadState as N, useAssetBulkDownload as V } from "./react/useAssetBulkDownload.es.js";
import { useAssetChooser as H } from "./react/useAssetChooser.es.js";
import { useAssetUpload as q } from "./react/useAssetUpload.es.js";
import { useAssetViewer as K } from "./react/useAssetViewer.es.js";
import { useBlockAssets as W } from "./react/useBlockAssets.es.js";
import { useBlockTemplates as Y } from "./react/useBlockTemplates.es.js";
import { useBlockSettings as $ } from "./react/useBlockSettings.es.js";
import { useBrandportalLink as oe } from "./react/useBrandportalLink.es.js";
import { useCategorizedDocumentPages as te } from "./react/useCategorizedDocumentPages.es.js";
import { useColorPalettes as pe } from "./react/useColorPalettes.es.js";
import { useColors as ue } from "./react/useColors.es.js";
import { useCoverPage as ae } from "./react/useCoverPage.es.js";
import { useDocumentCategories as xe } from "./react/useDocumentCategories.es.js";
import { useDocumentGroups as De } from "./react/useDocumentGroups.es.js";
import { useDocumentPageTargets as ce } from "./react/useDocumentPageTargets.es.js";
import { useDocumentSection as ye } from "./react/useDocumentSection.es.js";
import { useDocumentTargets as Ce } from "./react/useDocumentTargets.es.js";
import { useEditorState as de } from "./react/useEditorState.es.js";
import { useFileInput as Se } from "./react/useFileInput.es.js";
import { useFileUpload as Be } from "./react/useFileUpload.es.js";
import { useGroupedDocuments as ke } from "./react/useGroupedDocuments.es.js";
import { useGuidelineActions as Re } from "./react/useGuidelineActions.es.js";
import { useNavigationManager as we } from "./react/useNavigationManager.es.js";
import { usePageTemplateSettings as Ge } from "./react/usePageTemplateSettings.es.js";
import { usePrivacySettings as Oe } from "./react/usePrivacySettings.es.js";
import { useReadyForPrint as Fe } from "./react/useReadyForPrint.es.js";
import { useTemplateAssets as Me } from "./react/useTemplateAssets.es.js";
import { useThemeAssets as Ve } from "./react/useThemeAssets.es.js";
import { useThemeSettings as He } from "./react/useThemeSettings.es.js";
import { useTemplateChooser as qe } from "./react/useTemplateChooser.es.js";
import { useUncategorizedDocumentPages as Ke } from "./react/useUncategorizedDocumentPages.es.js";
import { useUngroupedDocuments as We } from "./react/useUngroupedDocuments.es.js";
import { usePlatformAppBridge as Ye } from "./react/usePlatformAppBridge.es.js";
import { getAppBridgeBlockStub as $e } from "./tests/AppBridgeBlockStub.es.js";
import { getAppBridgePlatformAppStub as oo } from "./tests/AppBridgePlatformAppStub.es.js";
import { getAppBridgeThemeStub as to } from "./tests/AppBridgeThemeStub.es.js";
import { AssetApiDummy as po } from "./tests/AssetApiDummy.es.js";
import { AssetChooserOptionsDummy as uo } from "./tests/AssetChooserOptionsDummy.es.js";
import { AssetChooserResultDummy as ao } from "./tests/AssetChooserResultDummy.es.js";
import { AssetDummy as xo } from "./tests/AssetDummy.es.js";
import { BrandportalLinkApiDummy as Do } from "./tests/BrandportalLinkApiDummy.es.js";
import { BrandportalLinkDummy as co } from "./tests/BrandportalLinkDummy.es.js";
import { BulkDownloadDummy as yo } from "./tests/BulkDownloadDummy.es.js";
import { ColorApiCreateDummy as Co, ColorApiDummy as To, ColorApiPatchDummy as Po } from "./tests/ColorApiDummy.es.js";
import { ColorCreateDummy as ho, ColorDummy as Bo, ColorPatchDummy as bo } from "./tests/ColorDummy.es.js";
import { ColorPaletteApiCreateDummy as Eo, ColorPaletteApiDummy as Ro, ColorPaletteApiPatchDummy as Uo } from "./tests/ColorPaletteApiDummy.es.js";
import { ColorPaletteCreateDummy as vo, ColorPaletteDummy as Go, ColorPalettePatchDummy as Lo } from "./tests/ColorPaletteDummy.es.js";
import { CoverPageApiDummy as jo } from "./tests/CoverPageApiDummy.es.js";
import { CoverPageDummy as Io } from "./tests/CoverPageDummy.es.js";
import { DocumentApiDummy as No } from "./tests/DocumentApiDummy.es.js";
import { DocumentCategoryApiDummy as zo } from "./tests/DocumentCategoryApiDummy.es.js";
import { DocumentCategoryDummy as _o } from "./tests/DocumentCategoryDummy.es.js";
import { DocumentDummy as Jo } from "./tests/DocumentDummy.es.js";
import { DocumentGroupApiDummy as Qo } from "./tests/DocumentGroupApiDummy.es.js";
import { DocumentGroupDummy as Xo } from "./tests/DocumentGroupDummy.es.js";
import { DocumentPageApiDummy as Zo, DocumentPageDuplicateApiDummy as $o } from "./tests/DocumentPageApiDummy.es.js";
import { DocumentPageDummy as or, DocumentPageDuplicateDummy as rr } from "./tests/DocumentPageDummy.es.js";
import { DocumentSectionApiDummy as mr } from "./tests/DocumentSectionApiDummy.es.js";
import { DocumentSectionDummy as sr } from "./tests/DocumentSectionDummy.es.js";
import { GuidelineSearchResultApiDummy as ir } from "./tests/GuidelineSearchResultApiDummy.es.js";
import { GuidelineSearchResultDummy as fr } from "./tests/GuidelineSearchResultDummy.es.js";
import { HttpUtilResponseDummy as nr } from "./tests/HttpUtilResponseDummy.es.js";
import { DocumentPageTargetsApiDummy as lr, DocumentTargetsApiDummy as cr, SingleTargetApiDummy as gr, UpdateTargetsApiDummy as yr } from "./tests/TargetsApiDummy.es.js";
import { DocumentPageTargetsDummy as Cr, DocumentTargetsDummy as Tr, UpdateTargetsDummy as dr } from "./tests/TargetsDummy.es.js";
import { TemplateApiLegacyDummy as Sr } from "./tests/TemplateApiLegacyDummy.es.js";
import { TemplateLegacyDummy as Br } from "./tests/TemplateLegacyDummy.es.js";
import { TemplateDummy as kr } from "./tests/TemplateDummy.es.js";
import { UserApiDummy as Rr } from "./tests/UserApiDummy.es.js";
import { UserDummy as wr } from "./tests/UserDummy.es.js";
import { withAppBridgeBlockStubs as Gr } from "./tests/withAppBridgeBlockStubs.es.js";
import { withAppBridgeThemeStubs as Or } from "./tests/withAppBridgeThemeStubs.es.js";
import { LinkSettingsDisplay as Fr, LinkSettingsIconPosition as Ir, LinkType as Mr } from "./types/Document.es.js";
import { DocumentPageVisibility as Vr } from "./types/DocumentPage.es.js";
import { FileExtension as Hr } from "./types/FileExtension.es.js";
import { FileExtensionSets as qr } from "./types/FileExtensionSets.es.js";
import { FileType as Kr } from "./types/FileType.es.js";
import { GuidelineSearchResultTypeMap as Wr } from "./types/GuidelineSearchResult.es.js";
import { AssetChooserObjectType as Yr, AssetChooserProjectType as Zr, TerrificEvent as $r } from "./types/Terrific.es.js";
import { Topic as ot } from "./types/Topic.es.js";
import { rgbObjectToRgbString as tt, rgbStringToRgbObject as mt } from "./utilities/color.es.js";
import { getDatasetByClassName as st, getDatasetByElement as ut } from "./utilities/dom.es.js";
import { getMimeType as at, mimeTypes as ft } from "./utilities/file.es.js";
import { generateRandomString as nt } from "./utilities/hash.es.js";
import { HttpClient as lt } from "./utilities/httpClient.es.js";
import { notify as gt } from "./utilities/notify.es.js";
import { compareObjects as At, convertObjectCase as Ct, isObject as Tt, mergeDeep as dt } from "./utilities/object.es.js";
import { SUBSCRIBE_TIMEOUT as St, subscribe as ht } from "./utilities/subscribe.es.js";
e();
export {
  m as AppBridgeCreateAsset,
  s as AppBridgePlatformApp,
  po as AssetApiDummy,
  N as AssetBulkDownloadState,
  Yr as AssetChooserObjectType,
  uo as AssetChooserOptionsDummy,
  Zr as AssetChooserProjectType,
  ao as AssetChooserResultDummy,
  xo as AssetDummy,
  Do as BrandportalLinkApiDummy,
  co as BrandportalLinkDummy,
  yo as BulkDownloadDummy,
  Co as ColorApiCreateDummy,
  To as ColorApiDummy,
  Po as ColorApiPatchDummy,
  ho as ColorCreateDummy,
  Bo as ColorDummy,
  Eo as ColorPaletteApiCreateDummy,
  Ro as ColorPaletteApiDummy,
  Uo as ColorPaletteApiPatchDummy,
  vo as ColorPaletteCreateDummy,
  Go as ColorPaletteDummy,
  Lo as ColorPalettePatchDummy,
  bo as ColorPatchDummy,
  jo as CoverPageApiDummy,
  Io as CoverPageDummy,
  No as DocumentApiDummy,
  zo as DocumentCategoryApiDummy,
  _o as DocumentCategoryDummy,
  Jo as DocumentDummy,
  Qo as DocumentGroupApiDummy,
  Xo as DocumentGroupDummy,
  Zo as DocumentPageApiDummy,
  or as DocumentPageDummy,
  $o as DocumentPageDuplicateApiDummy,
  rr as DocumentPageDuplicateDummy,
  lr as DocumentPageTargetsApiDummy,
  Cr as DocumentPageTargetsDummy,
  Vr as DocumentPageVisibility,
  mr as DocumentSectionApiDummy,
  sr as DocumentSectionDummy,
  cr as DocumentTargetsApiDummy,
  Tr as DocumentTargetsDummy,
  n as FetchError,
  Hr as FileExtension,
  qr as FileExtensionSets,
  Kr as FileType,
  ir as GuidelineSearchResultApiDummy,
  fr as GuidelineSearchResultDummy,
  Wr as GuidelineSearchResultTypeMap,
  lt as HttpClient,
  i as HttpClientError,
  nr as HttpUtilResponseDummy,
  f as InitializationError,
  Fr as LinkSettingsDisplay,
  Ir as LinkSettingsIconPosition,
  Mr as LinkType,
  St as SUBSCRIBE_TIMEOUT,
  gr as SingleTargetApiDummy,
  Sr as TemplateApiLegacyDummy,
  kr as TemplateDummy,
  Br as TemplateLegacyDummy,
  $r as TerrificEvent,
  l as TimeoutReachedError,
  ot as Topic,
  yr as UpdateTargetsApiDummy,
  dr as UpdateTargetsDummy,
  Rr as UserApiDummy,
  wr as UserDummy,
  g as ValidationError,
  h as closeAssetChooser,
  G as closeTemplateChooser,
  At as compareObjects,
  Ct as convertObjectCase,
  A as createAsset,
  R as downloadAsset,
  nt as generateRandomString,
  $e as getAppBridgeBlockStub,
  oo as getAppBridgePlatformAppStub,
  to as getAppBridgeThemeStub,
  T as getAssetResourceInformation,
  P as getCurrentUser,
  st as getDatasetByClassName,
  ut as getDatasetByElement,
  at as getMimeType,
  Tt as isObject,
  dt as mergeDeep,
  ft as mimeTypes,
  gt as notify,
  B as openAssetChooser,
  k as openAssetViewer,
  I as openConnection,
  w as openNavigationManager,
  j as openNewPublication,
  L as openTemplateChooser,
  tt as rgbObjectToRgbString,
  mt as rgbStringToRgbObject,
  ht as subscribe,
  V as useAssetBulkDownload,
  H as useAssetChooser,
  q as useAssetUpload,
  K as useAssetViewer,
  W as useBlockAssets,
  $ as useBlockSettings,
  Y as useBlockTemplates,
  oe as useBrandportalLink,
  te as useCategorizedDocumentPages,
  pe as useColorPalettes,
  ue as useColors,
  ae as useCoverPage,
  xe as useDocumentCategories,
  De as useDocumentGroups,
  ce as useDocumentPageTargets,
  ye as useDocumentSection,
  Ce as useDocumentTargets,
  de as useEditorState,
  Se as useFileInput,
  Be as useFileUpload,
  ke as useGroupedDocuments,
  Re as useGuidelineActions,
  we as useNavigationManager,
  Ge as usePageTemplateSettings,
  Ye as usePlatformAppBridge,
  Oe as usePrivacySettings,
  Fe as useReadyForPrint,
  Me as useTemplateAssets,
  qe as useTemplateChooser,
  Ve as useThemeAssets,
  He as useThemeSettings,
  Ke as useUncategorizedDocumentPages,
  We as useUngroupedDocuments,
  Gr as withAppBridgeBlockStubs,
  Or as withAppBridgeThemeStubs
};
