// @frontify/fondue/dist/components/Flyout/Flyout.es.js
import { jsx as e, jsxs as V, Fragment as Y } from "react/jsx-runtime";
import { watchModals as j } from "@react-aria/aria-modal-polyfill";
import { useButton as H } from "@react-aria/button";
import { useFocusRing as N, FocusScope as G } from "@react-aria/focus";
import { useOverlayTrigger as W, OverlayContainer as k } from "@react-aria/overlays";
import { mergeProps as q } from "@react-aria/utils";
import { useOverlayTriggerState as z } from "@react-stately/overlays";
import { FOCUS_STYLE as A } from "../../utilities/focusStyle.es.js";
import { merge as J } from "../../utilities/merge.es.js";
import { useRef as i, useEffect as K } from "react";
import { useContainScroll as Q } from "./hooks/useContainScroll.es.js";
import { useOverlayPositionWithBottomMargin as X } from "./hooks/useOverlayPositionWithBottomMargin.es.js";
import { Overlay as Z } from "./Overlay.es.js";
import { LegacyFlyoutFooter as $ } from "./FlyoutFooter.es.js";
const bo = "#eaebeb", ho = "10px";
var oo = /* @__PURE__ */ ((o) => (o.Top = "top", o.Bottom = "bottom", o.TopLeft = "top left", o.BottomLeft = "bottom left", o.TopRight = "top right", o.BottomRight = "bottom right", o.Right = "right", o.Left = "left", o))(oo || {});
const to = ({
  trigger: o,
  isTriggerDisabled: p = !1,
  decorator: a,
  onConfirm: g,
  onCancel: u,
  children: c,
  onOpenChange: d,
  isOpen: t = !1,
  title: T = "",
  badges: b = [],
  hug: h = !0,
  fitContent: v = !1,
  fixedHeader: y,
  fixedFooter: R,
  contentMinHeight: F,
  legacyFooter: L = !0,
  placement: w = "bottom left",
  offset: x,
  updatePositionOnContentChange: C = !1
}) => {
  const n = z({ isOpen: t, onOpenChange: d }), { toggle: D, close: O } = n, r = i(null), s = i(null), f = i(null), { isFocusVisible: l, focusProps: B } = N(), { triggerProps: E, overlayProps: I } = W(
    { type: "dialog" },
    n,
    r
  ), { positionProps: S } = X({
    triggerRef: r,
    overlayRef: s,
    scrollRef: f,
    isOpen: t,
    placement: w,
    offset: x,
    updatePositionOnContentChange: C
  }), { buttonProps: _, isPressed: P } = H(
    { onPress: () => D(), elementType: "div", isDisabled: p },
    r
  );
  K(() => {
    const U = j();
    return () => U();
  }, []), Q(s, { isDisabled: !t });
  const m = q(_, E, B, {
    "aria-label": "Toggle Flyout Menu"
  }), M = typeof o == "function" ? o(m, r, { isFocusVisible: l, isPressed: P }) : /* @__PURE__ */ e(
    "div",
    {
      ...m,
      ref: r,
      className: J([
        "tw-outline-none tw-rounded",
        h ? "tw-mx-3" : "tw-w-full",
        l && A
      ]),
      "data-test-id": "flyout-trigger",
      children: o
    }
  );
  return /* @__PURE__ */ V(Y, { children: [
    M,
    t && /* @__PURE__ */ e(k, { children: /* @__PURE__ */ e(G, { restoreFocus: !0, contain: !0, children: /* @__PURE__ */ e(
      Z,
      {
        title: T,
        badges: b,
        decorator: a,
        isOpen: t,
        onClose: O,
        overlayTriggerProps: I,
        positionProps: S,
        fixedHeader: y,
        fixedFooter: L ? /* @__PURE__ */ e($, { onConfirm: g, onCancel: u }) : R,
        ref: s,
        scrollRef: f,
        fitContent: v,
        contentMinHeight: F,
        children: c
      }
    ) }) })
  ] });
};
to.displayName = "FondueFlyout";
export {
  bo as FLYOUT_DIVIDER_COLOR,
  ho as FLYOUT_DIVIDER_HEIGHT,
  to as Flyout,
  oo as FlyoutPlacement
};
