// @frontify/fondue/dist/components/Dropdown/SelectMenu/SelectMenu.es.js
import { jsx as r } from "react/jsx-runtime";
import { getMenuItems as f, getKeyItemRecord as p } from "../../ActionMenu/Aria/helper.es.js";
import { useListBox as d } from "@react-aria/listbox";
import { merge as w } from "../../../utilities/merge.es.js";
import { useRef as y } from "react";
import { SelectMenuItem as b } from "./SelectMenuItem.es.js";
import { SelectMenuSection as M } from "./SelectMenuSection.es.js";
const S = ({
  menuBlocks: m,
  ariaProps: n,
  state: e,
  ariaLabel: i = "Select Menu",
  scrollable: s = !1
}) => {
  const c = f(m), u = p(c), l = y(null), { listBoxProps: a } = d(
    {
      ...n,
      "aria-label": i
    },
    e,
    l
  );
  return /* @__PURE__ */ r(
    "ul",
    {
      ...a,
      ref: l,
      className: w([
        "tw-list-none tw-p-0 tw-m-0 tw-bg-base tw-z-20 focus-visible:tw-outline-none",
        s && "tw-overflow-y-auto"
      ]),
      children: [...e.collection].map((o) => /* @__PURE__ */ r(M, { ariaLabel: o["aria-label"], children: [...o.childNodes].map((t) => /* @__PURE__ */ r(
        b,
        {
          state: e,
          node: t,
          menuItem: u[t.key]
        },
        t.key
      )) }, o.key))
    }
  );
};
S.displayName = "FondueSelectMenu";
export {
  S as SelectMenu
};
