// @frontify/fondue/dist/components/InputLabel/InputLabel.es.js
import { jsxs as u, jsx as t } from "react/jsx-runtime";
import { TooltipIcon as h } from "../TooltipIcon/TooltipIcon.es.js";
import { IconSize as v } from "../../foundation/Icon/IconSize.es.js";
import { merge as n } from "../../utilities/merge.es.js";
const g = ({
  children: e,
  htmlFor: r,
  required: w = !1,
  disabled: l = !1,
  clickable: m = !1,
  tooltip: i = [],
  bold: p,
  "data-test-id": o = "input-label"
}) => {
  const x = (Array.isArray(i) ? i : [i]).map((a, s) => ({
    key: s,
    ...a
  }));
  return /* @__PURE__ */ u(
    "div",
    {
      className: n([
        "tw-inline-flex tw-leading-4 tw-items-center tw-gap-1 tw-font-sans tw-text-s tw-max-w-full tw-min-w-0 tw-flex-initial",
        l ? "tw-text-text-disabled" : "tw-text-text-weak"
      ]),
      "data-test-id": `${o}-container`,
      children: [
        /* @__PURE__ */ t("div", { className: "tw-flex-1 tw-overflow-hidden tw-text-ellipsis tw-whitespace-nowrap", children: /* @__PURE__ */ t(
          "label",
          {
            htmlFor: r,
            className: n([
              "tw-select-none tw-max-w-full",
              p && "tw-font-medium",
              l || !m ? "hover:tw-cursor-not-allowed tw-pointer-events-none" : "hover:tw-cursor-pointer hover:tw-text-text group-hover:tw-text-text"
            ]),
            "data-test-id": o,
            title: typeof e == "string" ? e : "",
            children: e
          }
        ) }),
        w && /* @__PURE__ */ t("span", { "data-test-id": `${o}-required`, className: "tw-h-4 tw-text-m tw-text-text-negative", children: "*" }),
        x.map(({ key: a, triggerIcon: s, triggerStyle: c, hoverDelay: d = 100, ...f }) => /* @__PURE__ */ t("div", { className: "tw-leading-3", children: /* @__PURE__ */ t(
          h,
          {
            tooltip: { ...f, hoverDelay: d },
            iconSize: v.Size16,
            triggerIcon: s,
            triggerStyle: c
          }
        ) }, a))
      ]
    }
  );
};
g.displayName = "FondueInputLabel";
export {
  g as InputLabel
};
