// @frontify/guideline-blocks-settings/dist/index.es.js
import "./styles.css.es.js";
export * from "@frontify/sidebar-settings";
import { BlockInjectButton as g } from "./components/BlockInjectButton/BlockInjectButton.es.js";
import { BlockItemWrapper as l } from "./components/BlockItemWrapper/BlockItemWrapper.es.js";
import { DEFAULT_DRAGGING_TOOLTIP as u, DEFAULT_DRAG_TOOLTIP as f } from "./components/BlockItemWrapper/constants.es.js";
import { Attachments as s } from "./components/Attachments/Attachments.es.js";
import { DownloadButton as P } from "./components/DownloadButton/DownloadButton.es.js";
import { RichTextEditor as c } from "./components/RichTextEditor/RichTextEditor.es.js";
import { LinkPlugin as T, createLinkPlugin as E } from "./components/RichTextEditor/plugins/LinkPlugin/index.es.js";
import { BUTTON_PLUGIN as L, ButtonPlugin as R, ELEMENT_BUTTON as h, createButtonPlugin as A } from "./components/RichTextEditor/plugins/ButtonPlugin/createButtonPlugin.es.js";
import { withButton as C } from "./components/RichTextEditor/plugins/ButtonPlugin/withButton.es.js";
import { insertButton as F } from "./components/RichTextEditor/plugins/ButtonPlugin/transforms/insertButton.es.js";
import { submitFloatingButton as b } from "./components/RichTextEditor/plugins/ButtonPlugin/transforms/submitFloatingButton.es.js";
import { unwrapButton as D } from "./components/RichTextEditor/plugins/ButtonPlugin/transforms/unwrapButton.es.js";
import { upsertButton as N } from "./components/RichTextEditor/plugins/ButtonPlugin/transforms/upsertButton.es.js";
import { upsertButtonText as _ } from "./components/RichTextEditor/plugins/ButtonPlugin/transforms/upsertButtonText.es.js";
import { wrapButton as v } from "./components/RichTextEditor/plugins/ButtonPlugin/transforms/wrapButton.es.js";
import { getUrlFromEditor as W } from "./components/RichTextEditor/plugins/ButtonPlugin/utils/getUrl.es.js";
import { createButtonNode as Q } from "./components/RichTextEditor/plugins/ButtonPlugin/utils/createButtonNode.es.js";
import { triggerFloatingButton as q } from "./components/RichTextEditor/plugins/ButtonPlugin/utils/triggerFloatingButton.es.js";
import { triggerFloatingButtonEdit as J } from "./components/RichTextEditor/plugins/ButtonPlugin/utils/triggerFloatingButtonEdit.es.js";
import { triggerFloatingButtonInsert as Y } from "./components/RichTextEditor/plugins/ButtonPlugin/utils/triggerFloatingButtonInsert.es.js";
import { BlockButtonStyles as $ } from "./components/RichTextEditor/plugins/ButtonPlugin/utils/styles.es.js";
import { Custom1Plugin as rt } from "./components/RichTextEditor/plugins/TextStylePlugins/custom1Plugin.es.js";
import { Custom2Plugin as ot } from "./components/RichTextEditor/plugins/TextStylePlugins/custom2Plugin.es.js";
import { Custom3Plugin as pt } from "./components/RichTextEditor/plugins/TextStylePlugins/custom3Plugin.es.js";
import { Heading1Plugin as at } from "./components/RichTextEditor/plugins/TextStylePlugins/heading1Plugin.es.js";
import { Heading2Plugin as mt } from "./components/RichTextEditor/plugins/TextStylePlugins/heading2Plugin.es.js";
import { Heading3Plugin as xt } from "./components/RichTextEditor/plugins/TextStylePlugins/heading3Plugin.es.js";
import { Heading4Plugin as ft } from "./components/RichTextEditor/plugins/TextStylePlugins/heading4Plugin.es.js";
import { ImageCaptionPlugin as st } from "./components/RichTextEditor/plugins/TextStylePlugins/imageCaptionPlugin.es.js";
import { ImageTitlePlugin as Pt } from "./components/RichTextEditor/plugins/TextStylePlugins/imageTitlePlugin.es.js";
import { PARAGRAPH_CLASSES as ct, ParagraphMarkupElement as yt, ParagraphMarkupElementNode as Tt, ParagraphPlugin as Et, createParagraphPlugin as kt } from "./components/RichTextEditor/plugins/TextStylePlugins/paragraphPlugin.es.js";
import { QuoteMarkupElementNode as Rt, QuotePlugin as ht, createQuotePlugin as At } from "./components/RichTextEditor/plugins/TextStylePlugins/quotePlugin.es.js";
import { AllTextStylePlugins as Ct, AllTextStyles as Mt, TextStylePluginsWithoutImage as Ft, TextStylesWithoutImage as Ut } from "./components/RichTextEditor/plugins/TextStylePlugins/helpers.es.js";
import { BlockStyles as Gt, TextStyles as Dt } from "./components/RichTextEditor/plugins/styles.es.js";
import { getDefaultPluginsWithLinkChooser as Nt } from "./components/RichTextEditor/pluginPresets/defaultPluginsWithLinkChooser.es.js";
import { THEME_PREFIX as _t } from "./components/RichTextEditor/constants.es.js";
import { LinkSelector as vt } from "./components/Link/LinkSelector/LinkSelector.es.js";
import { LinkInput as Wt } from "./components/Link/LinkInput.es.js";
import { getLegacyUrl as Qt, getLinkFromEditor as Xt, getUrl as qt, getUrlFromLinkOrLegacyLink as zt } from "./components/Link/utils/getUrl.es.js";
import { relativeUrlRegex as Kt } from "./components/Link/utils/relativeUrlRegex.es.js";
import { isValidUrl as Zt, isValidUrlOrEmpty as $t } from "./components/Link/utils/url.es.js";
import { convertToRteValue as rr } from "./helpers/convertToRichTextValue.es.js";
import { customCoordinatesGetterFactory as or } from "./helpers/customCoordinatesGetterFactory.es.js";
import { hasRichTextValue as pr } from "./helpers/hasRichTextValue.es.js";
import { isDownloadable as ar } from "./helpers/isDownloadable.es.js";
import { mapAppBridgeColorPaletteToFonduePalette as mr, mapAppBridgeColorPalettesToFonduePalettes as lr } from "./helpers/mapColorPalettes.es.js";
import { addHttps as ur } from "./helpers/addHttps.es.js";
import { useAttachments as dr } from "./hooks/useAttachments.es.js";
import { useDndSensors as Sr } from "./hooks/useDndSensors.es.js";
import { getBackgroundSettings as Br } from "./settings/background.es.js";
import { getBorderSettings as yr } from "./settings/border.es.js";
import { getBorderRadiusSettings as Er, getBorderRadiusSlider as kr } from "./settings/borderRadius.es.js";
import { getExtendedBorderRadiusSettings as Rr } from "./settings/borderRadiusExtended.es.js";
import { getGutterSettings as Ar } from "./settings/gutter.es.js";
import { getMarginSettings as Cr, getMarginSlider as Mr } from "./settings/margin.es.js";
import { getMarginExtendedSettings as Ur } from "./settings/marginExtended.es.js";
import { getPaddingSettings as Gr, getPaddingSlider as Dr } from "./settings/padding.es.js";
import { getPaddingExtendedSettings as Nr } from "./settings/paddingExtended.es.js";
import { BorderStyle as _r, GutterSpacing as wr, Margin as vr, Padding as Vr, Radius as Wr, Security as jr, borderStyleMap as Qr, gutterSpacingStyleMap as Xr, marginStyleMap as qr, paddingStyleMap as zr, radiusStyleMap as Jr } from "./settings/types.es.js";
import { getSecurityDownloadableSetting as Yr, getSecurityGlobalControlId as Zr } from "./settings/securityDownloadable.es.js";
import { getSecurityGlobalControlSetting as te } from "./settings/securityGlobalControl.es.js";
import { isDark as ee } from "./utilities/color/isDark.es.js";
import { toHex8String as ne } from "./utilities/color/toHex8String.es.js";
import { toHexString as ie } from "./utilities/color/toHexString.es.js";
import { toRgbaString as ge } from "./utilities/color/toRgbaString.es.js";
import { setAlpha as le } from "./utilities/color/setAlpha.es.js";
import { toColorObject as ue } from "./utilities/color/toColorObject.es.js";
import { getReadableColor as de } from "./utilities/color/getReadableColor.es.js";
import { toShortRgba as Se } from "./utilities/color/toShortRgba.es.js";
import { moveItemInArray as Be } from "./utilities/moveItemInArray.es.js";
import { getBackgroundColorStyles as ye } from "./utilities/react/getBackgroundColorStyles.es.js";
import { getBorderStyles as Ee } from "./utilities/react/getBorderStyles.es.js";
import { getRadiusStyles as Le } from "./utilities/react/getRadiusStyles.es.js";
import { joinClassNames as he } from "./utilities/react/joinClassNames.es.js";
var r = /* @__PURE__ */ ((t) => (t.Main = "main", t.Basics = "basics", t.Layout = "layout", t.Style = "style", t.Security = "security", t.Targets = "targets", t))(r || {});
const o = (t) => t, n = (t) => t;
export {
  Ct as AllTextStylePlugins,
  Mt as AllTextStyles,
  s as Attachments,
  L as BUTTON_PLUGIN,
  $ as BlockButtonStyles,
  g as BlockInjectButton,
  l as BlockItemWrapper,
  Gt as BlockStyles,
  _r as BorderStyle,
  R as ButtonPlugin,
  rt as Custom1Plugin,
  ot as Custom2Plugin,
  pt as Custom3Plugin,
  u as DEFAULT_DRAGGING_TOOLTIP,
  f as DEFAULT_DRAG_TOOLTIP,
  P as DownloadButton,
  h as ELEMENT_BUTTON,
  wr as GutterSpacing,
  at as Heading1Plugin,
  mt as Heading2Plugin,
  xt as Heading3Plugin,
  ft as Heading4Plugin,
  st as ImageCaptionPlugin,
  Pt as ImageTitlePlugin,
  Wt as LinkInput,
  T as LinkPlugin,
  vt as LinkSelector,
  vr as Margin,
  ct as PARAGRAPH_CLASSES,
  Vr as Padding,
  yt as ParagraphMarkupElement,
  Tt as ParagraphMarkupElementNode,
  Et as ParagraphPlugin,
  Rt as QuoteMarkupElementNode,
  ht as QuotePlugin,
  Wr as Radius,
  c as RichTextEditor,
  r as Sections,
  jr as Security,
  _t as THEME_PREFIX,
  Ft as TextStylePluginsWithoutImage,
  Dt as TextStyles,
  Ut as TextStylesWithoutImage,
  ur as addHttps,
  Qr as borderStyleMap,
  rr as convertToRteValue,
  Q as createButtonNode,
  A as createButtonPlugin,
  E as createLinkPlugin,
  kt as createParagraphPlugin,
  At as createQuotePlugin,
  or as customCoordinatesGetterFactory,
  o as defineBlock,
  n as defineSettings,
  ye as getBackgroundColorStyles,
  Br as getBackgroundSettings,
  Er as getBorderRadiusSettings,
  kr as getBorderRadiusSlider,
  yr as getBorderSettings,
  Ee as getBorderStyles,
  Nt as getDefaultPluginsWithLinkChooser,
  Rr as getExtendedBorderRadiusSettings,
  Ar as getGutterSettings,
  Qt as getLegacyUrl,
  Xt as getLinkFromEditor,
  Ur as getMarginExtendedSettings,
  Cr as getMarginSettings,
  Mr as getMarginSlider,
  Nr as getPaddingExtendedSettings,
  Gr as getPaddingSettings,
  Dr as getPaddingSlider,
  Le as getRadiusStyles,
  de as getReadableColor,
  Yr as getSecurityDownloadableSetting,
  Zr as getSecurityGlobalControlId,
  te as getSecurityGlobalControlSetting,
  qt as getUrl,
  W as getUrlFromEditor,
  zt as getUrlFromLinkOrLegacyLink,
  Xr as gutterSpacingStyleMap,
  pr as hasRichTextValue,
  F as insertButton,
  ee as isDark,
  ar as isDownloadable,
  Zt as isValidUrl,
  $t as isValidUrlOrEmpty,
  he as joinClassNames,
  mr as mapAppBridgeColorPaletteToFonduePalette,
  lr as mapAppBridgeColorPalettesToFonduePalettes,
  qr as marginStyleMap,
  Be as moveItemInArray,
  zr as paddingStyleMap,
  Jr as radiusStyleMap,
  Kt as relativeUrlRegex,
  le as setAlpha,
  b as submitFloatingButton,
  ue as toColorObject,
  ne as toHex8String,
  ie as toHexString,
  ge as toRgbaString,
  Se as toShortRgba,
  q as triggerFloatingButton,
  J as triggerFloatingButtonEdit,
  Y as triggerFloatingButtonInsert,
  D as unwrapButton,
  N as upsertButton,
  _ as upsertButtonText,
  dr as useAttachments,
  Sr as useDndSensors,
  C as withButton,
  v as wrapButton
};
