// @frontify/fondue/dist/components/Tree/types.es.js
const e = {
  none: "tw-my-0",
  "x-small": "tw-my-0.5",
  small: "tw-my-1",
  medium: "tw-my-1.5",
  large: "tw-my-2",
  "x-large": "tw-my-2.5"
}, t = {
  none: "tw-shadow-none",
  "x-small": "tw-shadow-sm",
  small: "tw-shadow",
  medium: "tw-shadow-md",
  large: "tw-shadow-lg",
  "x-large": "tw-shadow-xl"
}, o = {
  none: "tw-rounded-none",
  "x-small": "tw-rounded-sm",
  small: "tw-rounded",
  medium: "tw-rounded-md",
  large: "tw-rounded-lg",
  "x-large": "tw-rounded-xl"
}, r = {
  none: "tw-border-0",
  "x-small": "tw-border",
  small: "tw-border-2",
  medium: "tw-border-4",
  large: "tw-border-8"
}, l = {
  none: "tw-border-none",
  solid: "tw-border-solid",
  dotted: "tw-border-dotted",
  dashed: "tw-border-dashed"
}, d = {
  none: {
    textColor: "",
    selectedTextColor: "",
    backgroundColor: "",
    selectedBackgroundColor: "",
    pressedBackgroundColor: "",
    dragHanlderTextColor: "",
    selectedDragHanlderTextColor: ""
  },
  neutral: {
    textColor: "tw-text-text",
    selectedTextColor: "tw-font-medium tw-text-box-neutral-strong-inverse",
    backgroundColor: "group-hover:tw-bg-box-neutral",
    selectedBackgroundColor: "tw-bg-box-neutral-strong group-hover:tw-bg-box-neutral-strong-hover",
    pressedBackgroundColor: "group-active:tw-bg-box-neutral-pressed",
    dragHanlderTextColor: "tw-text-text",
    selectedDragHanlderTextColor: "tw-text-box-neutral-strong-inverse"
  },
  soft: {
    textColor: "tw-text-text",
    selectedTextColor: "tw-font-medium tw-text-box-neutral-strong",
    backgroundColor: "group-hover:tw-bg-box-neutral-hover",
    selectedBackgroundColor: "tw-bg-box-neutral-strong-inverse-hover group-hover:tw-bg-box-neutral-strong-inverse-hover",
    pressedBackgroundColor: "group-active:tw-bg-box-neutral-pressed",
    dragHanlderTextColor: "tw-text-text",
    selectedDragHanlderTextColor: "tw-text-box-neutral-strong"
  }
};
export {
  r as TreeItemBorderClassMap,
  o as TreeItemBorderRadiusClassMap,
  l as TreeItemBorderStyleClassMap,
  d as TreeItemColorsClassMap,
  t as TreeItemShadowClassMap,
  e as TreeItemSpacingClassMap
};
