// @frontify/fondue/dist/components/Popper/Popper.es.js
import { jsx as n, Fragment as g } from "react/jsx-runtime";
import { createContext as D, useState as p, useEffect as O, useMemo as R, Children as v, isValidElement as I, useContext as A } from "react";
import { usePopper as H } from "react-popper";
import { Portal as L } from "../Portal/Portal.es.js";
const N = 200, S = 400, U = "100px", y = D({}), s = ({ children: t }) => t;
s.displayName = "FonduePopperReference";
const i = ({ children: t }) => {
  const { open: r } = G();
  return r ? /* @__PURE__ */ n(g, { children: t }) : null;
};
i.displayName = "FonduePopperContent";
const G = () => {
  const t = A(y);
  if (!t)
    throw new Error("Popper compound components cannot be rendered outside the Popper component");
  return t;
}, c = ({
  children: t,
  open: r,
  placement: a = "bottom-start",
  offset: m = [0, 8],
  flip: u = !0,
  enablePortal: w = !0,
  zIndex: d = "auto",
  isDetached: l = !1,
  verticalAlignment: x
}) => {
  const [C, F] = p(null), [T, f] = p(null), [P, _] = p({
    width: N,
    height: S
  }), e = H(l ? document.body : C, T, {
    placement: a,
    modifiers: [
      { name: "offset", options: { offset: m } },
      {
        name: "flip",
        enabled: u
      }
    ]
  });
  O(() => {
    (async () => {
      e.update && (await e.update(), e.state && _({
        width: e.state.rects.popper.width,
        height: e.state.rects.popper.height
      }));
    })().catch(console.error);
  }, [u, a, m, r]);
  const h = l ? {
    left: `${(window.innerWidth - P.width) / 2}px`,
    top: x === "top" ? U : `${(window.innerHeight - P.height) / 2}px`,
    transform: "none"
  } : {}, b = R(() => ({ open: r }), [r]);
  return /* @__PURE__ */ n(y.Provider, { value: b, children: v.map(t, (o) => {
    if (I(o) && typeof o.type == "function") {
      const { name: E } = o.type;
      if (E === s.name)
        return /* @__PURE__ */ n("div", { ref: F, children: o });
      if (E === i.name)
        return w ? /* @__PURE__ */ n(L, { children: /* @__PURE__ */ n(
          "div",
          {
            ref: f,
            style: {
              zIndex: d,
              ...e.styles.popper,
              ...h
            },
            ...e.attributes.popper,
            children: o
          }
        ) }) : /* @__PURE__ */ n(
          "div",
          {
            ref: f,
            style: { zIndex: d, ...e.styles.popper, ...h },
            ...e.attributes.popper,
            children: o
          }
        );
    }
  }) });
};
c.displayName = "FonduePopper";
c.Reference = s;
c.Content = i;
export {
  c as Popper
};
