// @frontify/guideline-blocks-settings/dist/components/RichTextEditor/plugins/LinkPlugin/FloatingLink/InsertLinkModal/InsertModal.es.js
import { jsxs as l, jsx as e } from "react/jsx-runtime";
import { FormControl as p, TextInput as x, Button as n, ButtonSize as r, ButtonStyle as k, ButtonEmphasis as b, IconCheckMark20 as f } from "@frontify/fondue";
import { LinkInput as v } from "../../../../../Link/LinkInput.es.js";
const I = ({
  state: t,
  onTextChange: o,
  onUrlChange: d,
  onToggleTab: a,
  onCancel: c,
  onSave: w,
  isValidUrlOrEmpty: i,
  hasValues: u,
  testId: m,
  appBridge: h,
  children: s
}) => /* @__PURE__ */ l("div", { "data-test-id": m, className: "tw-bg-white tw-rounded tw-shadow tw-p-7 tw-min-w-[400px] tw-overflow-y-auto", children: [
  /* @__PURE__ */ e(
    p,
    {
      label: {
        children: "Text",
        htmlFor: "linkText",
        required: !0
      },
      children: /* @__PURE__ */ e(x, { id: "linkText", value: t.text, placeholder: "Link Text", onChange: o })
    }
  ),
  s,
  /* @__PURE__ */ e("div", { className: "tw-mt-5", children: /* @__PURE__ */ e(
    v,
    {
      url: t.url,
      newTab: t.newTab,
      onUrlChange: d,
      onToggleTab: a,
      isValidUrlOrEmpty: i,
      appBridge: h
    }
  ) }),
  /* @__PURE__ */ e("div", { className: "tw-mt-3", children: /* @__PURE__ */ l("div", { className: "tw-pt-5 tw-flex tw-gap-x-3 tw-justify-end tw-border-t tw-border-t-black-10", children: [
    /* @__PURE__ */ e(
      n,
      {
        onClick: c,
        size: r.Medium,
        style: k.Default,
        emphasis: b.Default,
        children: "Cancel"
      }
    ),
    /* @__PURE__ */ e(
      n,
      {
        onClick: w,
        size: r.Medium,
        icon: /* @__PURE__ */ e(f, {}),
        disabled: !u || !i(t == null ? void 0 : t.url),
        children: "Save"
      }
    )
  ] }) })
] });
export {
  I as InsertModal
};
